<html>

<head>
    <meta content="text/html; charset=UTF-8" http-equiv="content-type">
    <link rel="stylesheet" href="/style-document.css">
</head>

<body>
    <div id="top" class="body">
        <h1 style="text-decoration: underline; text-align: center">DJMol v2.1 Windows-64 version Document</h1>
        <div id="toc_container">
            <p class="toc_title">Table of Contents</p>
            <ul class="toc_list">
                <li><a href="#chapter-1">Chapter 1: Introduction</a></li>
                <li><a href="#chapter-2">Chapter 2: DFTB+ Calculations</a></li>
                <li><a href="#chapter-3">Chapter 3: Modeling with Siesta</a></li>
                <li><a href="#chapter-4">Chapter 4: OpenMD Simulations</a></li>
                <li><a href="#chapter-5">Chapter 5: Python scripting with ASE</a></li>
                <li><a href="#chapter-6">Chapter 6: Tools of DJMol</a></li>
                <li><a href="#chapter-7">Chapter 7: Architecture of the Program</a></li>
                <li><a href="#chapter-8">Chapter 8: Software Applications</a></li>
                <li><a href="#appendices">Appendices</a></li>
                <ul class="toc_list">
                    <li><a href="#appendix-a">Python Scripting for DJMol Applications</a></li>
                    <li><a href="#appendix-b">Installation of Python and NumPy</a></li>
                    <li><a href="#appendix-c">Windows Subsystem Linux in DJMol</a></li>
                    <li><a href="#appendix-d">Saved Data</a></li>
                    <li><a href="#appendix-e">Demonstration Videos</a></li>
                    <li><a href="#appendix-f">For Compiling/Installation of DJMol and Add-Ons</a></li>
                    <li><a href="#appendix-g">Forthcoming Features</a></li>
                </ul>
            </ul>
        </div>
        <div id="chapter-1">
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c58 c66 c41 c76">1</span></p>
            <p class="c79 c13 title"><span class="c29">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Introduction</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c30 c13 c7"><span class="c1 c58 c146"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c42">DJMol is an ongoing open source (licensed with GPL v3 or later) &lsquo;</span><span class="c42 c66">Modeling Platform</span><span class="c6">&rsquo; project and it is architectured for performing various simulations in computational materials science / chemistry. It is a free, object-oriented software developed by IKST, (Indo-Korea Science and Technology Centre, 17/1, Bellary Rd, Yashoda Nagar, Yelahanka, Bengaluru, Karnataka 560064), Bangalore.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">In its core level, DJMol is integrated with widely accepted and open source (GPL/LGPL ) computational software, such as,</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c42">DFTB+ (University of Bremen, Germany), the </span><span class="c42 c66">default calculator</span><span class="c6">&nbsp;of DJMol</span></p>
            <p class="c2 c13"><span class="c6">Atomic Simulation Environment (ASE, Technical University of Denmark)</span></p>
            <p class="c2 c13"><span class="c6">SIESTA (Institut de Ciencia de Materials de Barcelona (ICMAB), Spain)</span></p>
            <p class="c2 c13"><span class="c6">OpenMD (University of Notre Dame., USA).</span></p>
            <p class="c2 c13"><span class="c42">The main target of this software is the researchers (including experimentalists) and academicians and industrial researchers. In addition to this, a site, </span><span class="c42 c66">https://sites.google.com/view/djmolplatfor</span>
                <span class="c66">m</span><span class="c66 c55 c41">&nbsp;</span><span class="c42">is created for more technical details, updates and demonstration videos/notes etc. The code-development resources are recently included in its official site, </span>
                <span class="c42 c66">www.djmol.info</span>
                <span class="c6">.</span>
            </p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">As mentioned before, the software is integrated with DFTB+ (as its default calculator), and Siesta (for higher level DFT calculations), along with ASE and OpenMD,. Hence a short review on these packages is added for a quick-reference.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c13 c20"><span class="c80 c58 c55 c41">Computational Tools</span></p>
            <p class="c20 c13 c7"><span class="c80 c58 c55 c41"></span></p>
            <p class="c2 c13"><span class="c6">Here, a short overview of the four main components of the platform is described.</span></p>
            <p class="c2 c13"><span class="c4">DFTB+: </span><span class="c6">DFTB (Density functional tight binding) method is an electronic structure method which is specially designed to model large molecular/crystal/nano-systems. The origin of this method is routed in DFT itself, which is, arguably, the most popular </span>
                <span class="c0">ab initio</span><span class="c6">&nbsp;method in computational materials science or in quantum chemistry. It is developed in 80&rsquo;s and it has been introduced into the computational chemistry area mainly by work of Prof. Helmut Eschrig, Prof. Gotthard Seifert, and Prof. Thomas Frauenheim et. al. Since it uses DFT parameters, DFTB is also known as </span>
                <span class="c0">approximate</span><span class="c6">&nbsp;DFT method.</span></p>
            <p class="c21 c13"><span class="c6">This method uses atomic parameters (as it is stored in the SK file set, a text file) which are generated from highly accurate atomic DFT calculations using PBE functional. It also consists of diatomic repulsive potentials from B3LYP functional. Note that DFTB+ uses minimal valence basis set (numerical Slater type AOs) and one has to explicitly specify a particular SK data file for a given calculation. </span></p>
            <p class="c13 c154"><span class="c6">A list of SK </span><span class="c4">file set</span><span class="c6">&nbsp;list is given below (see</span><span class="c0">, www.dftb.org</span><span class="c6">), which is by default, added in the DJMol suite. Other SK file which are not included (such as, &lt;3ob:freq&gt; set which requires &lt;3ob&gt;) should manually incorporate into their input scripts (it is freely available); For more details, see: </span></p>
            <p class="c13 c157"><span class="c0">https://www.dftbplus.org/</span><span class="c6">&nbsp;</span><span class="c0">and </span></p>
            <p class="c25 c13"><span class="c0">https://www.dftb.org/parameters/download/</span><span class="c6">&nbsp;</span></p>
            <p class="c75 c13 c7"><span class="c6"></span></p>
            <p class="c75 c13 c7"><span class="c6"></span></p>
            <p class="c21 c13 c7"><span class="c6"></span></p>
            <p class="c21 c13 c7"><span class="c27 c36"></span></p>
            <p class="c21 c13"><span class="c27 c36">Table 1</span><span class="c27 c11">: Set of common SK data set </span></p>
            <p class="c94 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 495.20px; height: 206.07px;"><img alt="" src="../images/image65.png" style="width: 495.20px; height: 206.07px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c2 c13"><span class="c4">SIESTA: </span><span class="c6">Spanish Initiative for Electronic Simulations with Thousands of Atoms is a fast (since it uses linear scaling DFT methods) density functional theoretical code which uses pseudopotentials. It can handle molecular as well as extended systems (metals, polymers, defects in the metals, surfaces etc.) It is a free code and for more details see: </span>
                <span class="c0">https://departments.icmab.es/leem/siesta</span><span class="c6">. And </span><span class="c0">routinely </span><span class="c6">it can be used to predict:</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c33 c13"><span class="c6">&bull; Total and partial energies.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Atomic forces.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Stress tensor.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Electric dipole moment.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Atomic, orbital and bond populations (Mulliken).</span></p>
            <p class="c33 c13"><span class="c6">&bull; Electron density.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Geometry relaxation, fixed or variable cell.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Constant-temperature molecular dynamics (Nose thermostat).</span></p>
            <p class="c33 c13"><span class="c6">&bull; Variable cell dynamics (Parrinello-Rahman).</span></p>
            <p class="c33 c13"><span class="c6">&bull; Spin polarized calculations (collinear or not).</span></p>
            <p class="c33 c13"><span class="c6">&bull; k-sampling of the Brillouin zone.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Local and orbital-projected density of states.</span></p>
            <p class="c33 c13"><span class="c6">&bull; COOP and COHP curves for chemical bonding analysis.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Dielectric polarization.</span></p>
            <p class="c33 c13"><span class="c6">&bull; Vibrations (phonons).</span></p>
            <p class="c33 c13"><span class="c6">&bull; Band structure.</span></p>
            <p class="c2 c13"><span class="c46 c41">Note that at present TranSiesta (electron transport module) is not supported by DJMol</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">Since Siesta uses </span><span class="c0">Numerical Atomic Orbital </span><span class="c6">(i.e. localized basis set) as its basis, some familiarity with these sets are highly appreciated. Note that a user can give &lsquo;Basis set&rsquo; information to Siesta in two different ways. The most simple method is to provide a &lsquo;keyword&rsquo;of the basis set. See the next table for its details:</span></p>
            <p class="c21 c13"><span class="c27 c36">Table 2</span><span class="c27 c11">: A set of common basis set in Siesta </span></p>
            <a id="t.d90e2e382e663ecd9dbb0c38e98fde33886bcf98"></a>
            <a id="t.0"></a>
            <table class="c169">
                <tbody>
                    <tr class="c167">
                        <td class="c100" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">Single-Zeta</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                        <td class="c158" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">Double-Zeta</span></p>
                        </td>
                        <td class="c140" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">Triple-Zeta</span></p>
                        </td>
                        <td class="c72" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">Single-Zeta</span></p>
                            <p class="c20"><span class="c6">Polarized</span></p>
                        </td>
                        <td class="c72" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">Double-Zeta</span></p>
                            <p class="c20"><span class="c6">Polarized</span></p>
                        </td>
                        <td class="c72" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">Triple-Zeta</span></p>
                            <p class="c20"><span class="c6">Polarized</span></p>
                        </td>
                        <td class="c118" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">Triple-Zeta</span></p>
                            <p class="c20"><span class="c6">Double-</span></p>
                            <p class="c20"><span class="c6">Polarized</span></p>
                        </td>
                    </tr>
                    <tr class="c124">
                        <td class="c100" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">SZ</span></p>
                        </td>
                        <td class="c158" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">DZ</span></p>
                        </td>
                        <td class="c140" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">TZ</span></p>
                        </td>
                        <td class="c72" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">SZP</span></p>
                        </td>
                        <td class="c72" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">DZP</span></p>
                        </td>
                        <td class="c72" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">TZP</span></p>
                        </td>
                        <td class="c118" colspan="1" rowspan="1">
                            <p class="c20 c7"><span class="c6"></span></p>
                            <p class="c20"><span class="c6">TZDP</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c6">However, for advanced calculations a user can also give the basis set data explicitly. Refer Siesta manual for its description.</span></p>
            <p class="c2 c13"><span class="c6">Apart from the basis set, one should also select an appropriate pseudopotential (PP). For more details, see:</span></p>
            <p class="c2 c13"><span class="c0">[1]https://departments.icmab.es/leem/siesta/Pseudopotentials/index.html or,</span></p>
            <p class="c2 c13"><span class="c0">[2]https://departments.icmab.es/leem/siesta/Databases/Pseudopotentials/periodictable-gga-abinit.html</span></p>
            <p class="c2 c13"><span class="c6">The first link redirects to ATOM program which can be used to generate PP for elements within the GGA/LDA functionals. The second link gives a database of PP which was converted from the Abinit PP. Note that choosing PP is a crucial step in Siesta calculation so a user must be Very Careful at this step. </span></p>
            <p class="c2 c13"><span class="c4">Note</span><span class="c6">: </span><span class="c46 c41">By default, DJMol use PP from the Second link [2].</span></p>
            <p class="c2 c13 c7"><span class="c46 c41"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 413.93px; height: 267.35px;"><img alt="" src="../images/image67.png" style="width: 413.93px; height: 267.35px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c30 c13"><span class="c27 c36">Figure 1.1</span><span class="c27 c11">: The pseudopotentials of Abinit which is converted for Siesta (figure courtesy: ICMAB)</span></p>
            <p class="c2 c13"><span class="c46 c41">A note on Functionals</span><span class="c6">: In Siesta you can specify either LDA or GGA type functionals in a &lt;.fdf&gt; input file. For example, see below for a part of the &lt;fdf&gt; file, which define functionals :</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c21 c13"><span class="c27 c36">Table 3</span><span class="c27 c11">: Specefication of functional in Siesta FDF file. </span></p>
            <a id="t.bd410d79a5625115026a7d03e37059a69e1f3213"></a>
            <a id="t.1"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c87" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&hellip;</span></p>
                            <p class="c2"><span class="c4">XC.functional</span><span class="c6">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;LDA &nbsp; &nbsp; &nbsp; &nbsp; # Exchange-correlation functional ( GGA can be also used)</span></p>
                            <p class="c2"><span class="c4">XC.authors &nbsp; </span><span class="c6">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;CA &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# Particular parametrization of xc func.</span></p>
                            <p class="c2"><span class="c6">SpinPolarized &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; .false. &nbsp; &nbsp; &nbsp; # Spin unpolarized calculation</span></p>
                            <p class="c2"><span class="c6">...</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c21 c13"><span class="c27 c36">Table 4</span><span class="c27 c11">: Set of </span><span class="c6">choice of </span><span class="c4">XC.authors</span><span class="c6">&nbsp;(with its type) variables.</span></p>
            <a id="t.00ebe9e78af3e370716ce450310d36212e881408"></a>
            <a id="t.2"></a>
            <table class="c82">
                <tbody>
                    <tr class="c156">
                        <td class="c31" colspan="1" rowspan="1">
                            <p class="c33"><span class="c6">&nbsp;CA (equivalent to PZ) &nbsp; &nbsp; &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;PW92 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;PW91 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;PBE &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;revPBE &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;RPBE &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;WC &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;AM05 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;PBEsol &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;PBEJsJrLO &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;BLYP (equivalent to LYP) &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;DRSLL (equivalent to DF1) &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;LMKLL (equivalent to DF2) &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;KBM &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;C09 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;BH &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                            <p class="c33"><span class="c6">&nbsp;VV &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span></p>
                        </td>
                        <td class="c171" colspan="1" rowspan="1">
                            <p class="c33"><span class="c6">&nbsp;A LDA type</span></p>
                            <p class="c33"><span class="c6">&nbsp;LDA </span></p>
                            <p class="c33"><span class="c6">&nbsp;GGA </span></p>
                            <p class="c33"><span class="c6">&nbsp;GGA </span></p>
                            <p class="c33"><span class="c6">&nbsp;Modified GGA-PBE &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;Modified GGA-PBE &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;Modified GGA-PBE &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;Modified GGA-PBE </span></p>
                            <p class="c33"><span class="c6">&nbsp;Modified GGA-PBE &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;GGA-PBE &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;GGA &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;van der Waals density functional (vdW-DF) &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;vdW-DF functional &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;vdW-DF functional &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;vdW-DF functional &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;vdW-DF functional &nbsp;</span></p>
                            <p class="c33"><span class="c6">&nbsp;vdW-DF functional &nbsp;</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">For a complete specification on &lt;fdf&gt; input format please read the Siesta Manual (v 4.0 or greater). The DJMol, is currently equipped with a SIESTA 4.1 version (with OpenMP support) and a number of its pre - as well as post-processing tools. </span></p>
            <p class="c2 c13"><span class="c4">ASE: </span><span class="c6">Atomic Simulation Environment - is a Python based LGPL library which comprises a variety of post processing tools or methods. A general outline of ASE is following (block diagram courtesy: Dr. A H Larsen)</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 272.29px; height: 364.29px;"><img alt="" src="../images/image66.png" style="width: 272.29px; height: 384.86px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 1.2</span><span class="c27 c11">: Different modules in ASE. Interface denotes the DJMol platform.</span></p>
            <p class="c2 c13"><span class="c6">One should note that to use ASE, a minimum-level knowledge on Python programming language (version 3.x) is a mandate. See Appendix-A for a basic course on Python programming. &nbsp;ASE&rsquo;s input should written in an object-oriented style, so that the syntax of the input is highly flexible and it can be used to perform a series of jobs, by using, say, a for-loop. Another advantage is, it can be effectively coupled with other python base libraries such as NumPy, Matplotlib etc. to create different types of output files or data set.</span></p>
            <p class="c2 c13"><span class="c6">Most useful tasks with ASE are: structure optimization (with or without constraints), molecular dynamics, and performing nudged elastic band calculations or its variants. For more details see: </span><span class="c0">https://wiki.fysik.dtu.dk/ase/</span>
                <span class="c6">.</span>
            </p>
            <p class="c2 c13"><span class="c4">OpenMD</span><span class="c6">: it is an MD tool based on force-field type potential and it is widely used for performing MD simulations of </span><span class="c0">open </span><span class="c6">systems (such as &nbsp;liquids, proteins, nanoparticles, interfaces, and other complex systems). Note that we used 2.6 version of the program. The code is written in C++ and uses Python for its data pre/post processing. In the DJMol CygWin based C++ binaries are used and it is kept in ./OpenMD folder with other relevant parameter-files, Python scripts etc.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">The OpenMD manual is accompanied with the DJMol and it is available at:</span></p>
            <p class="c2 c13"><span class="c0">http://openmd.org/wp-content/docs/OpenMD-2.6.pdf</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c4">OpenMD</span><span class="c6">&nbsp;uses a range of programs and Python scripts and a user must familiarize with its uses before the production calculation. Examples of OpenMD is illustrated in: </span><span class="c0">http://openmd.org/category/examples/</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c27 c36">Table 5</span><span class="c27 c11">: Important binaries of OpenMD and its descriptions.</span></p>
            <a id="t.f11b07e6fca66ba154feb35eeb8845f2a6d90f3e"></a>
            <a id="t.3"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;openmd.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">The Main OpenMD Executable for MD run.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;atom2omd.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">atom2omd attempts to construct .omd files from files containing only atomic coordinate information</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;Dump2XYZ.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Converts an OpenMD dump file into a file suitable for viewing in a molecular dynamics viewer.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;DynamicProps.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Computes time correlation functions like the velocity autocorrelation function, or the mean square displacement.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;elasticConstants.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">elasticConstants computes the general elastic constants that relate stress and strain for a given input configuration.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;Hydro.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Hydro generates hydrodynamic resistance tensor (.hydro) files which are required when using the Langevin integrator using complex rigid bodies.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;icosahedralBuilder.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">icosahedralBuilder creates single-component geometric solids that can be useful in simulating nanostructures.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">nanoparticleBuilder.exe, nanorodBuilder.exe,nanorod_</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">programs to construct nanoparticles of various sizes and geometries.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">omd2omd.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">omd2omd is a utility script which helps in replicating, rotating, and translating already built OpenMD .omd, .dump, and.eor files.</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;randomBuilder.exe, simpleBuilder.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">generate crystals.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;recenter.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">recenter is a utility script which moves all integrable objects in an OpenMD file so that the center of mass is at the origin.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">SequentialProps.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Computes a time history of static properties from a dump file.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;StaticProps.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Computes static properties like the pair distribution function.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">&nbsp;thermalizer.exe</span></p>
                        </td>
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Thermalizer gives the atoms some initial velocities (at a given temperature) before the &nbsp;equilibration.</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c21 c13"><span class="c27 c36">Table 4</span><span class="c6">: Short Descriptions the Python scripts.</span></p>
            <a id="t.89ad73e8c2d52a72ea684b41bcced062d26c8db1"></a>
            <a id="t.4"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">affineScale.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">OpenMD affine scaling transform</span></p>
                            <p class="c2"><span class="c6">Takes an OpenMD file and scales both the periodic box and the coordinates of all StuntDoubles in the system by the same amount.</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                            <p class="c2"><span class="c6">You can either specify a new volume scaling for isotropic scaling, or specify one (or more) of the coordinates for non-isotropic scaling.</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">dumpConverter.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Dump File Converter Converts old-style OOPSE md and dump files into new OpenMD style combined files</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">funcflExtractor.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Funcfl Extractor Opens an DYNAMO86 EAM funcfl file, parses the data and writes out separate files for F[rho], rho(r), and Z(r)</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2 c7"><span class="c6"></span></p>
                            <p class="c2"><span class="c6">hbtetAnalyzer.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2 c7"><span class="c6"></span></p>
                            <p class="c2"><span class="c6">It is used for doing analysis on Hydrogen Bond Tetrahedrality matrices</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">hydroExplainer.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Computes predicted diffusion constants and rotational relaxation times from a hydro file. &nbsp;Explains the values in the hydro file in terms of properties that can be calculated from a molecular dynamics simulation.</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">lcorrzFit.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">A utility script to analyze of Legendre correlation functions </span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">omdLast.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">OpenMD File Extractor: Makes omd file from the last good frame of OpenMD dump file.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">omdShrink.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">OpenMD File Trimmer: Skips every n frames of an OpenMD dump file and loads it into new dump file</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">omd-solvator.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">OMD Solvator: Opens two omd files, one with a solute structure and one with a solvent structure. &nbsp;Deletes any solvent molecules that overlap with solute molecules and produces a new combined omd file. &nbsp;The output omd</span></p>
                            <p class="c2"><span class="c6">file must be edited to run properly in OpenMD. &nbsp;Note that the two boxes must have identical box geometries (specified on the Hmat line).</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">omdSplit.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">OpenMD File Splitter: Splits OpenMD dump file frames into separate omd files</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">pack2omd.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Packmol RigidBody Replacer: Finds atomistic rigid bodies in a packmol-generated xyz file and generates an OpenMD (omd) file with center of mass and orientational coordinates for rigid bodies.</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">principalAxisCalculator.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">It opens an XYZ file and computes the moments of inertia and principal axes for the structure in the XYZ file. &nbsp;Optionally rotates the structure so that the long axis (that with the smallest eigenvalue) is pointing along the z-axis.</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">protonSampler.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">This python script will generate proton disordered configurations of ice-Ih, given an input of the (.xyz) coordinates of the oxygen positions in the lattice.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">slabBuilder.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">SlabBuilder to create starting omd files for arbitrary surface cuts specified by Miller indices (hkl) of FCC, BCC and SC materials </span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">slipLength.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">slipLength is a built in analysis script which can compute the slip-length of a solid-liquid interface under shear. The script assumes the solid is placed in the middle of the box, with equal amounts of liquid on either side.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">solLiqFricCalc.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Used for calculating solid liquid friction coefficients</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">solvator.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">A script that reads in a water box and solute xyz (or pdb), merges the two systems, and</span></p>
                            <p class="c2"><span class="c6">deletes overlapping molecules</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">stat2dielectric.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">A script that computes the static dielectric constant.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">stat2dipolecorr.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">A script that computes the system dipole correlation function</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">stat2tension.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Used for computing surface tensions from pressure tensors in stat files</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">stat2thcond.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Computes the correlation function of the heat flux vector that has been stored in a stat file. &nbsp; These can be used to compute the thermal conductivity.</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">stat2visco.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Computes various correlation functions of the pressure and pressure tensor that have been stored in a stat file. &nbsp; These can be used to compute shear and bulk viscosities.</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">vcorr2spectrum.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">A script that processes a velocity autocorrelation function into a amplitude spectrum Post-processes a vcorr file and creates a normalized spectrum.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">waterBoxer.py </span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6 c178">builds a lattice of water molecules in a periodic box </span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">waterReplacer.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Finds atomistic waters in an xyz file and generates an OpenMD (omd) file with center of mass and orientational coordinates for rigid body waters.</span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">waterRotator.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">Samples water orientations from a list of known good orientations</span></p>
                            <p class="c2 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c24" colspan="1" rowspan="1">
                            <p class="c33 c13"><span class="c6">wcorr2spectrum.py</span></p>
                        </td>
                        <td class="c14" colspan="1" rowspan="1">
                            <p class="c2"><span class="c6">A script that processes a charge velocity autocorrelation function into a amplitude spectrum ;Post-processes a wcorr file and creates a normalized spectrum.</span></p>
                            <p class="c33 c13 c7"><span class="c6"></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <h2 class="c148 c13" id="h.gjdgxs"><span class="c80 c58 c77 c86"></span></h2>
            <p class="c20 c13"><span class="c80 c58 c55 c41">Major Components of the Software</span></p>
            <p class="c2 c13"><span class="c42">DJMol program is, in essence, a rich client platform (RCP) GUI application that has been built on the top of the RCP framework of the Netbeans IDE. The underlying OOP paradigm permits simple and straight-forward addition and extension of extra modules or plug-ins into its RCP frame work. This application can also be viewed as a </span>
                <span class="c42 c66">molecular</span><span class="c42">&nbsp;</span><span class="c42 c66">workbench</span><span class="c6">&nbsp;that integrates many application programs for modeling molecular systems into a single framework. The main components of the GUI are discussed below.</span></p>
            <ol class="c112 lst-kix_list_5-0 start" start="1">
                <li class="c18 c13 c93 c59 li-bullet-0"><span class="c6">Basic IDE components: As indicated earlier the IDE component, inherited from Netbeans RCP module, is the core part the user-interface. The IDE framework offers a standard set of tools dedicated for advanced programming or scripting tasks. IDE significantly improve productivity of its users since it automates several text based processes (code completion, version control, project management, restoring old versions of the text files etc.). Several of native IDE menus are integrated; e.g. the </span>
                    <span class="c0">File</span><span class="c6">&nbsp;menu selects a coordinate data by its file extension (eg. xyz or hsd) to visualize the structure and open its text to the text editor of the IDE. Note that in the current version of the program xyz or hsd files are selected as the default file extensions for the viewer. The selected file can be manipulated at different levels by using its </span>
                    <span class="c0">Edit</span><span class="c6">, </span><span class="c0">View</span><span class="c6">&nbsp;and </span><span class="c0">Source</span><span class="c6">&nbsp;menus. Other basic IDE features like refactoring of files, Git facility, visual </span>
                    <span class="c0">diff</span><span class="c6">&nbsp;utility (to compare the contents of two text files) etc. are also supported. Project Explorer - which is placed on the far left side - shows the packages (directories and it files) that make up Python projects; It can also be used to systematically arrange or manipulate other data. </span>
                    <span class="c0">Files</span><span class="c6">&nbsp;and </span><span class="c0">Favorite</span><span class="c6">&nbsp;explorer panels are two tools for managing files (including binaries) in general. See Fig A for the IDE text editor which displays a Python script.</span></li>
            </ol>
            <p class="c2 c13 c7 c93"><span class="c63 c42 c58"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 510.66px; height: 188.54px;"><img alt="" src="../images/image69.png" style="width: 510.66px; height: 207.34px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c30 c13"><span class="c36">Figure 1.3</span><span class="c27 c11">: A Python script is displayed in the IDE text editor with syntax highlighting option.</span></p>
            <ol class="c112 lst-kix_list_5-0" start="2">
                <li class="c18 c13 c93 c59 li-bullet-0"><span class="c6">Top Component Window: The center panel of the software is set to display molecules and it embeds an unabridged Jmol visualizer binary and it is used in the </span><span class="c0">TopComponent</span><span class="c6">&nbsp;class (the basic unit of RCP display). Optionally a </span>
                    <span class="c0">scripting</span><span class="c6">&nbsp;tool (for Jmol scripting) can be invoked from this panel to create more sophisticated and customized molecular visualizations (See Table 5).</span><span class="c3">&nbsp; </span></li>
            </ol>
            <p class="c18 c13 c7 c93"><span class="c27 c36"></span></p>
            <p class="c18 c13 c7 c93"><span class="c27 c36"></span></p>
            <p class="c18 c13 c7 c93"><span class="c27 c36"></span></p>
            <p class="c18 c13 c7 c93"><span class="c27 c36"></span></p>
            <p class="c18 c13 c7 c93"><span class="c27 c36"></span></p>
            <p class="c18 c13 c7 c93"><span class="c27 c36"></span></p>
            <p class="c2 c13 c7 c93"><span class="c27 c36"></span></p>
            <p class="c2 c13"><span class="c36">Table 5</span><span class="c27 c11">: Some selected one-liner commands of Jmol scripts and its descriptions.</span></p>
            <a id="t.68785223501d2820bde09dd7b2793509c656cc19"></a>
            <a id="t.5"></a>
            <table class="c44">
                <tbody>
                    <tr class="c172">
                        <td class="c104" colspan="1" rowspan="1">
                            <p class="c33 c13"><span class="c5">load =cod/1000373 {2,2,2}</span></p>
                            <p class="c33 c13 c7"><span class="c5"></span></p>
                            <p class="c33 c13"><span class="c5">minimize</span></p>
                            <p class="c33 c13 c7"><span class="c5"></span></p>
                            <p class="c33 c13 c7"><span class="c5"></span></p>
                            <p class="c33 c13"><span class="c5">write &quot;C:\\file.pdb&quot;</span></p>
                            <p class="c33 c13 c7"><span class="c5"></span></p>
                            <p class="c33 c13"><span class="c5">isosurface cutoff 0.01 &quot;C:\\cube.gz&quot;</span></p>
                            <p class="c33 c13 c7"><span class="c5"></span></p>
                            <p class="c33 c13"><span class="c34">q=quaternion()</span></p>
                        </td>
                        <td class="c138" colspan="1" rowspan="1">
                            <p class="c2 c13"><span class="c42">Loads 2x2x2 supercell of NaVO</span><span class="c42 c62">2</span><span class="c42">F</span><span class="c42 c62">2</span><span class="c6">&nbsp;from COD database</span></p>
                            <p class="c2 c13"><span class="c6">Molecular mechanics (UFF) steepest descent minimization</span></p>
                            <p class="c2 c13 c7"><span class="c6"></span></p>
                            <p class="c2 c13"><span class="c6">Save the molecule into a PDB file</span></p>
                            <p class="c2 c13 c7"><span class="c6"></span></p>
                            <p class="c2 c13"><span class="c6">Visualize an isosurface from a gzipped CUBE file</span></p>
                            <p class="c2 c13 c7"><span class="c6"></span></p>
                            <p class="c2 c13"><span class="c6">Saving orientation in quaternion rather than Euler angles</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c18 c13 c7 c93"><span class="c3"></span></p>
            <ol class="c112 lst-kix_list_5-0" start="3">
                <li class="c18 c13 c93 c59 li-bullet-0"><span class="c6">2D and 3D Data Windows: JFreeChart library (a Java library for displaying various types of scientific data) is used to create a 2D graphs and it is usually projected onto a </span><span class="c0">JPanel</span><span class="c6">&nbsp;(a Swing GUI widget toolkit) to display the plots. By this, one can add more control options to the rendered graphs (See, Fig-D for a 2D plot with more control options). It is the </span>
                    <span class="c0">default</span><span class="c6">&nbsp;2D data plot library of the DJMol and it has several graphic styles and it offers many configuration options to customize the rendering of graphics and it export the graphic content in PNG format. Apart from this, Matplotlib library is also used to display 2D graphics especially with Siesta add-on. The 3D graphics (scalar volumetric data such as density isosurface) windows all are embedded with Jmol application with some internal Jmol script commands. To manipulate the 3D data either Jmol scripts or Pythons scripts are used. For example, a Python script is used to convert xsf file format (which is frequently used in Siesta) into the cube format. </span></li>
                <li class="c18 c13 c59 c93 li-bullet-0"><span class="c6">Terminal Window: One can run OS commands and Python scripts with this window. Note that in the Windows version of the program the CygWin environment is automatically linked with this terminal. If necessary, WSL (a compatibility layer for running native Linux executables under Windows 10 OS) can also be linked. See the manual for more details. The plug-in for the Python integration into the software can be switched to get Python 2.x as well as Python 3.x. It gives the terminal based (interactive mode) access to the language. As an example, by using this terminal one can install many additional packages like ASE or Matplotlib library. Using SSH commands of CygWin or WSL this terminal can be connected with other remote machines.</span></li>
                <li class="c18 c13 c93 c59 li-bullet-0"><span class="c6">Console Window: This window is usually placed immediately below the Top Component window and it is mainly used to display console output text (eg. to show DFTB+ output when it runs). It also contains Start and Stop buttons to execute the DFTB+ program. The log text from this window can be saved or post processed, optionally.</span></li>
                <li class="c2 c13 c93 c59 li-bullet-0"><span class="c6">Add-on Windows: Other major window components are from the add-on programs based on Swing API. Currently there are six add-ons are constructed and all these programs are independent of each other. And these add-ons are built from </span>
                    <span class="c0">JFrame</span><span class="c6">&nbsp;&ndash; the base container of the application. At present it supports all the AWT (abstract window toolkit) components. Although more recent </span><span class="c0">JavaFX</span>
                    <span class="c6">&nbsp;class to create desktop applications exist in Java, we always used Swing based GUIs since more libraries and tools are available for Swing API.</span>
                </li>
            </ol>
            <p class="c2 c13"><span class="c6">Schematically all these key GUI components are shown in the figure 1.4 .</span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 539.35px; height: 323.88px;"><img alt="" src="../images/image68.png" style="width: 545.49px; height: 323.88px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c2 c13"><span class="c36">Figure 1.4</span><span class="c27 c11">: A schematic classification of major window components which constitute the entire GUI of DJMol program.</span></p>
        </div>
        <div id="chapter-2">
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c66 c41 c108">2</span></p>
            <p class="c79 c13 title"><span class="c29">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; DFTB+ Calculations</span></p>
            <h2 class="c13 c148"><span class="c98 c36 c58"></span></h2>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c6">Here we illustrate some standard procedures that can be performed by the DFTB+ program in conjunction with DJMol platform. At present there are three distinct file types can be opened for DFTB+ calculations, viz., [1] dftb_in.hsd, &lt;.xyz&gt; and &lt;.gen&gt; files. The first one is the default unique file type for the program and the &nbsp;last two file extensions can be converted into &lt;dftb_in.hsd&gt; file by using a script writer (see below section).</span></p>
            <p class="c33 c13 c7"><span class="c6"></span></p>
            <a id="t.5413fcf34d3142013ff074bf49f11edecdc224c7"></a>
            <a id="t.6"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container" id="h.30j0zll"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c109">[1] DFTB+ Script Writer</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c9"><span class="c6">If one opens a &lt;.xyz&gt; or &lt;.hsd&gt; file, for visualizing the structure, subsequently one can also try to make a corresponding &lt;dftb_in.hsd&gt; file using a DFTB+ script writer program. Using this, a user can make a basic &lt;dftb_in.hsd&gt; script for a range of jobs, including, geometry optimization, calculating excited states or Hessian etc. It can be retrieved by:</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6 c10">Set Up </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;DFTB Scripting </span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">And the generated &lt;dftb_in.hsd&gt; file is kept in &lt;</span><span class="c4">Input</span><span class="c6">&gt; folder. Note that the atomic information including its geometry is directly taken from &lt;gen&gt; data of the currently loaded file.</span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">Note that, the Script-Writer module will write &lt;TEMPdftb_in.hsd&gt; into the &lt;Input&gt; folder. It is strongly recommended that, a user should always try to inspect/modify this script before its submission, if it is necessary.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 306.99px; height: 202.14px;"><img alt="" src="../images/image70.png" style="width: 306.99px; height: 202.14px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c94 c13" id="h.1fob9te"><span class="c27 c36">Figure 2.1</span><span class="c27 c11">: A screen shot of the script writer tool for the DFTB+ program.</span></p>
            <h3 class="c75 c13 c99" id="h.3znysh7"><span class="c42 c58 c63"></span></h3>
            <a id="t.cae73925200a674a105533d7dc22570fc51547d5"></a>
            <a id="t.7"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c162">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c73 c42">[2] Executing DFTB+ Scripts</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c9"><span class="c6">Running DFTB+ calculator is possible when you load a &lt;dftb_in.hsd&gt; file. Make sure that the &lt;Input&gt; folder should have this file before the execution and this file can be executed as:</span></p>
            <p class="c2 c9 c7"><span class="c6 c10"></span></p>
            <p class="c2 c9"><span class="c6 c10">Execute </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Run DFTB+</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c57"><span class="c6">And if necessary, this Run can be killed by</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6 c10">Execute </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Stop DFTB+</span></p>
            <p class="c2 c9" id="h.2et92p0"><span class="c6">The generated/loaded files are usually placed in the` &lt;DFTB_Scratch&gt;filder. And these files are also sent back as a Zip file, to the directory where your current &lt;hsd&gt; file resides.</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c6">DFTB+ generated data can be analyzed or visualized by the program. Main data analyzers are given below.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <a id="t.497b9299816e828f648a8b01334cc9404c8afbda"></a>
            <a id="t.8"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container" id="h.tyjcwt"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c162">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c73 c42">[3] Force Components</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c9"><span class="c6">DFTB+ out file contains information about Cartesian force components (in atomic units) and this can be visualized as a line chart as shown below. The &lt;detailed.out&gt; file can be opened by:</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c13 c57"><span class="c6 c10">Tools </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp; ForceComponents</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 342.62px; height: 193.48px;"><img alt="" src="../images/image74.png" style="width: 342.62px; height: 193.48px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c2 c13"><span class="c27 c36">Figure 2.2</span><span class="c27 c11">: The </span><span class="c61 c36 c58 c66">x</span><span class="c27 c11">&nbsp;components of the Cartesian forces of the Fullerene molecule after the geometry optimization. The shaded portion can be easily zoomed in for more details.</span></p>
            <a id="t.2df6dd721d274bc5b311c420756969810c7dec08"></a>
            <a id="t.9"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c81">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c109">[4] Molecular Orbitals and Energy Levels</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c9 c7" id="h.3dy6vkm"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">MO information (including density and density difference files) are stored in the .CUBE format. In order to get cube files one needs to ensure that dftb_in.hsd file contains relevant keywords (</span><span class="c0">WriteDetailedXML</span>
                <span class="c6">&nbsp;and </span><span class="c0">WriteEigenvectors</span><span class="c6">, see example directory for a template file). After running this file it will produce 3 out files (detailed.xml, charges.bin, and eigenvec.bin) and these files should be transferred from &lt;DFTB_Scratch&gt; directory to &lt;ModesBinary&gt; folder. &nbsp;Then invoke:</span></p>
            <p class="c2 c13 c7 c57"><span class="c6 c10"></span></p>
            <p class="c2 c13 c57"><span class="c6 c10">Execute </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Run Waveplot</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">From this window, select the proper SK file set (look at the combo box entitled, </span><span class="c0">Specify SK set and Generate HSD file</span><span class="c6">) and press on the button &lt;</span><span class="c4">Generate Waveplot File</span>
                <span class="c6">&gt;. This will produce the needed HSD file for the Waveplot.exe binary file. View this file from the second tab and edit it if needed. Then invoke the &lt;</span><span class="c4">Calculate</span><span class="c6">&gt; button to produce the necessary CUBE file. Note that it might take some minutes to complete the task (depending on the size of the system).</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">To visualize the cubes file, first copy the Full Path of the folder, &lt;ModesBinary&gt; and then go to:</span></p>
            <p class="c2 c13 c7"><span class="c6 c10"></span></p>
            <p class="c2 c13 c57"><span class="c6 c10">Tools </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;View Cube</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">Then click on the button, &lt;</span><span class="c4">Open Directory</span><span class="c6">&gt; and insert the recently copied Full Path into it and select a CUBE file. This will list all the CUBE files into the table, and this can be visualized easily by selecting the needed file name from the table and by clicking on the &lt;</span>
                <span class="c4">Open</span><span class="c6">&gt; button. Optionally one can also use </span><span class="c0">JMol scripting </span><span class="c6">facility for more advanced file visualizations or manipulations.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 540.64px; height: 356.99px;"><img alt="" src="../images/image73.png" style="width: 540.64px; height: 356.99px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 2.3:</span><span class="c27 c11">&nbsp;Visual of one of the MOs from the list of CUBE files.</span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">To view only MO energy levels and its degeneracies, one can use the &lt;detailed.out&gt; file and this can be called by:</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c57"><span class="c6 c10">Tools </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp; MO Levels</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">The green lines represent occupied and red lines represent the un occupied levels. For accurate calculations (which use tight SCC parameters) one can also look at the degeneracies levels.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 371.33px; height: 440.00px;"><img alt="" src="../images/image75.png" style="width: 371.33px; height: 440.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 2.4:</span><span class="c27 c11">&nbsp;MO energy levels from the detailed.out file.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <a id="t.94c9b4decc43011203b1785816b13984011c4d6c"></a>
            <a id="t.10"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c153">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c109">[5] UV-Visible Spectrum Convolution</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7" id="h.1t3h5sf"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">DFTB+ uses TD-DFTB for calculating oscillator strength of the optical excitations and this is stored in </span><span class="c4">SPX.dat</span><span class="c6">&nbsp;file. An add-on is used for reading this data and this can be invoked by:</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c13 c57"><span class="c6 c10">Tools </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp; UV-Vis Spectrum</span></p>
            <p class="c2 c13 c7"><span class="c6 c10"></span></p>
            <p class="c2 c9"><span class="c6">Select a </span><span class="c4">SPX.dat</span><span class="c6">&nbsp;file to proceed. This will plot the oscillator strengths (vertical red lines) and its fitted function ( broadened through convolution with a Gaussian, Lorentzian function or its linear combination, known as pseudo-Voigt functions - shown by blue curves). By default it uses </span>
                <span class="c0">eV </span><span class="c6">however one can also obtain an equivalent spectrum using </span><span class="c0">nm </span><span class="c6">units (this creates a separate plot). Note that the oscillator strength is </span>
                <span class="c0">dimensionless </span>
                <span class="c6">quantity and it is described in the </span><span class="c0">y </span><span class="c6">axis.</span></p>
            <p class="c2 c13 c7"><span class="c6 c10"></span></p>
            <p class="c2 c9"><span class="c6">By right clicking the plot, one can adjust several parameters (Auto Range, Zoom, Grid Lines, Axes properties etc.) of the figure and can save the image into PNG format.</span></p>
            <p class="c2 c7 c13"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 549.26px; height: 210.16px;"><img alt="" src="../images/image76.png" style="width: 549.26px; height: 210.16px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure </span><span class="c27 c11">2.5: UV-Visible spectrum of an organic molecule fitted with Lorentzian convolution.</span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <a id="t.2ab7ab9c8567fcce615504efa3128c30d7cef3b9"></a>
            <a id="t.11"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container" id="h.4d34og8"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c180">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c13 c25"><span class="c73 c42">[6] Partial Charges</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">DFTB+ caclulates partial charges and this charges can be projected out into the individual atoms. This Utility can be invoked by:</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c13 c57"><span class="c6 c10">Tools </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Partial Charges &nbsp;</span></p>
            <p class="c2 c13 c7"><span class="c6 c10"></span></p>
            <p class="c2 c9"><span class="c6">This utility read both the XYZ geometry and its corresponding </span><span class="c0">detailed.out</span><span class="c6">&nbsp;file. A color scale will also be shown for a reference. Note that this will generate a file and this file be saved for a reference (See Appendix section for the </span>
                <span class="c4">Saved Data</span><span class="c6">).</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 455.18px; height: 287.36px;"><img alt="" src="../images/image77.png" style="width: 455.18px; height: 287.36px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 2.6</span><span class="c27 c11">: Numerical values of Mulliken&#39;s partial charges are represented on atoms</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <a id="t.d52e611ee8e5ad780eb520838246f1e6e35c19a0"></a>
            <a id="t.12"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c164">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c109">[7] Molecular Vibrations</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">By using Harmonic approximation DFTB+ calculates the vibrational normal modes and frequencies. This requires Hessian Matrix (calculated by the DFTB+) and a script file for modes.exe binary (which calculates modes and frequencies).</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">The first step is to optimize the geometry. And once the optimized geometry is ready one can do the </span><span class="c0">vibrational calculation</span><span class="c6">&nbsp;with the necessary keywords (it includes Driver = SecondDerivatives{} keywords; See the example directory). This will generate the needed </span>
                <span class="c0">hessian.out </span><span class="c6">file for the </span><span class="c0">modes.exe. </span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">To calculate vibrational modes vectors, to visualize/animate these modes and to get the frequencies, invoke the tool as:</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6 c10">Execute &#10148; Run Modes </span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">Here in this tool one has to enter the optimized GEN styled geometry </span></p>
            <p class="c2 c9 c7"><span class="c3"></span></p>
            <p class="c23 c13"><span class="c27 c41 c45">Geometry= GenFormat {</span></p>
            <p class="c23 c13"><span class="c27 c41 c45">&nbsp; ...</span></p>
            <p class="c23 c13"><span class="c27 c41 c45">}</span></p>
            <p class="c20 c9 c7"><span class="c5"></span></p>
            <p class="c2 c9"><span class="c6">and the hessian data as:</span></p>
            <p class="c2 c13 c7 c121"><span class="c5"></span></p>
            <p class="c2 c13 c120"><span class="c27 c41 c45">hessian = {</span></p>
            <p class="c2 c13 c120"><span class="c27 c41 c45">...</span></p>
            <p class="c2 c13 c120"><span class="c27 c41 c45">}</span></p>
            <p class="c2 c9"><span class="c6">Note that the dots (...) represents the numerical value of the Hessian matrix ie. ALL the numerical value of the file , hessian.out, which is </span><span class="c4">3Nx3N</span><span class="c6">&nbsp;matrix data (force constant matrix).</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">After these two data entry select the corresponding SK file set and then click on the button: </span><span class="c46 c41">Generate Script. </span><span class="c6">This will generate modes_in.hsd file in the ModesBinary folder. This file contains all the information for the modes.exe program which </span>
                <span class="c0">diagonalizes </span><span class="c6">the hessian matrix to give modes/frequency information. The button, </span><span class="c4">Generate Modes/Frequencies </span><span class="c6">initiate this operation and produces the file, modes.xyz which contains the frequency/modes information.</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">To view this file another application is used this can be called by:</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6 c10">Tools </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;View Modes</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">This will read the files modes.xyz and visualize the vectors/animations etc. Note that the Hessian data is used by the modes.exe binary </span><span class="c0">as such</span><span class="c6">. In other words it does not projects out translational and rotational motions. So that the user should be optimize the molecule well enough. If needed see </span>
                <span class="c0">http://gaussian.com/vib/</span><span class="c6">&nbsp;on the projection technique of Hessian matrix (Sayvetz conditions).</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 547.60px; height: 347.52px;"><img alt="" src="../images/image78.png" style="width: 547.60px; height: 347.52px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c30 c13"><span class="c27 c36">Figure 2.7</span><span class="c27 c11">: Vibrational modes of water molecules read from the file modes.xyz in the ModesBinary directory.</span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <a id="t.53ead40ea2083836423716044ff7e8fdeca9cc3c"></a>
            <a id="t.13"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c81">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c109">[8] Real Time Molecular Dynamics</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <h3 class="c75 c13 c99" id="h.2s8eyo1"><span class="c103 c42 c58"></span></h3>
            <p class="c2 c9"><span class="c6">MD simulations can be done with or without using this tool. Normally one can submit MD job as usual (like a static calculation job). However, this MD tool is mainly used in real time so that a user can analyze Trajectory properties (Total Energy, Kinetic/Potential Energies, &nbsp;structure of each step) at Run Time. Moreover FFT based autocorrelation functions can be used to calculate IR spectra (from the dipole moments or from the velocity autocorrelation functions) after finishing the MD run. Note that, the MD analyzer tool should be invoked </span>
                <span class="c0">before</span><span class="c6">&nbsp;the DFTB+ calculation. While running the MD calculation it will show</span></p>
            <p class="c2 c9"><span class="c6">the progress of the data along with the current structure. Its step are:</span></p>
            <p class="c33 c13 c78"><span class="c6">[1] Remove </span><span class="c41 c46">all </span><span class="c6">the files from ./DFTB_Scratch directory. It is Mandate.</span></p>
            <p class="c33 c13 c78"><span class="c6">[2] Open MD file as, &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c6 c10">File </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Open (hsd)</span></p>
            <p class="c33 c13 c78"><span class="c6">[3] Invoke the MD tool as, &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c6 c10">Tools </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;MD &nbsp;Analyzer</span></p>
            <p class="c33 c13 c78"><span class="c6">[4] Execute MD run as, &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c6 c10">Execute </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Run DFTB+</span></p>
            <p class="c33 c13 c78"><span class="c6">[5] Wait for the MD calculation to finish; and after that use tools like, &lt;Velocity Autocorrelation Spectra&gt;.</span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 453.83px; height: 237.24px;"><img alt="" src="../images/image58.png" style="width: 453.83px; height: 254.68px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span>
                <span class="c6">&nbsp;</span>
            </p>
            <p class="c20 c13"><span class="c27 c36">Figure</span><span class="c27 c11">&nbsp;2.8: Variations of energies during the MD, in real time.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <a id="t.546bea6b78f8aeb5eab3b655ea3e7434e2fbda74"></a>
            <a id="t.14"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13" id="h.17dp8vu image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c149">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c73">[9] Potential Energy Surfaces (2D/1D)</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">The DJMol creates 2D PES (potential energy surface) by performing a series of DFTB+ calculations. For example, PES of H</span><span class="c19">2</span><span class="c6">O</span><span class="c19">2</span><span class="c6">&nbsp;(hydrogen peroxide) molecule is created by doing a batch process. To create 2D PES we need two independent internal coordinate variables and, arbitrarily, we have chosen the O-O bond length and the H-H dihedral angle (in degrees) as shown in Fig. X.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 314.83px; height: 216.96px;"><img alt="" src="../images/image59.png" style="width: 314.83px; height: 216.96px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c30 c13"><span class="c27 c36">Figure 2.9</span><span class="c27 c11">: The two independent variables of H</span><span class="c32 c11">2</span><span class="c27 c11">O</span><span class="c32 c11">2</span><span class="c27 c11">&nbsp; PES. The unit of bond length is in Angstrom and that of the dihedral angle is in degrees.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">In first, make the structure of the molecule (or see Example folder for its coordinates); then using the &lt;Convert Structure&gt; tool (See Fig.2), convert this XYZ formatted file into the MOPIN format (which contains an equivalent z-matrix data). And copy the content of this file into &lt;Build PES HSD&gt; tool as shown in the Fig. 3.</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">By clicking the button, &lt;Generate PES HSD Files&gt;, this will generate 441 HSD input files for the DFTB+ program (it will take around ten minutes); copy all of these HSD files into the &lt;SCRATCH&gt; folder (a user can also select any other directory instead of this one) and then invoke the &lt;Batch Processing Tool&gt;. This will start DFTB+ program to execute all these files and create the PES in the &lt;PESScan&gt; folder. </span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 461.71px; height: 259.89px;"><img alt="" src="../images/image60.png" style="width: 461.71px; height: 259.89px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c2 c13"><span class="c27 c11">&nbsp;</span><span class="c27 c36">Figure 2.10</span><span class="c27 c11">: The Convert tool is used to transform Cartesian file into a Mopin file (which essentially contain the z-matrix in Mopac&rsquo;s format).</span></p>
            <p class="c2 c13 c7"><span class="c27 c11"></span></p>
            <p class="c20 c13 image-container"><span class="c6">&nbsp;</span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 464.07px; height: 322.89px;"><img alt="" src="../images/image61.png" style="width: 464.07px; height: 322.89px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c27 c36">Figure 2.11</span><span class="c27 c11">: The converted MOPIN files content is inserted in the below text area and insert the variable name in the appropriate position. VariableI represents the O-O bond length (starts from 1.2&Aring;) and VariableII, it H-H dihedral angle (starts with 0.0</span>
                <span class="c28 c11">o</span><span class="c27 c11">).</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">After finishing the batch process, &lt;PESViewer&gt; can be invoked to get the plot of 2D PES or its contour diagram as shown in the Fig. 4. This utility will also indicates the minimum energy data point on the PES (in other words from this point one can start geometry optimization to locate local minimum) and its corresponding geometry file.</span></p>
            <p class="c2 c13"><span class="c6">In &lt;PESScan&gt; folder one will see the following data files (See Table 6).</span></p>
            <p class="c2 c13"><span class="c36">Table 6</span><span class="c27 c11">: Datafiles of PES calculations.</span></p>
            <a id="t.114ea66389a2b94896a843353cd5ffc675e2b420"></a>
            <a id="t.15"></a>
            <table class="c82">
                <tbody>
                    <tr class="c122">
                        <td class="c134" colspan="1" rowspan="1">
                            <p class="c84 c13"><span class="c0">View2DPES.dat</span></p>
                            <p class="c84 c13"><span class="c0">&nbsp;</span></p>
                            <p class="c84 c13"><span class="c0">2DPESscan.dat</span></p>
                            <p class="c13 c84"><span class="c0">&nbsp;</span></p>
                            <p class="c84 c13"><span class="c0">2DPESscanmovie.xyz</span></p>
                        </td>
                        <td class="c125" colspan="1" rowspan="1">
                            <p class="c84 c13"><span class="c0">Gives the data intervals in x and y directions and energy as E(x,y) in eV</span></p>
                            <p class="c84 c13"><span class="c0">It contains the sorted file names used in the PES creation with energy (in eV and in Hartree).</span></p>
                            <p class="c84 c13"><span class="c0">Contains the geometry of the individual molecules in xyz format. It can be visualized/animated with DJMol. This can be userd to construct PES with other ab initio programs.</span></p>
                        </td>
                    </tr>
                    <tr class="c159">
                        <td class="c176" colspan="1" rowspan="1">
                            <p class="c13 c83 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 170.42px; height: 176.26px;"><img alt="" src="../images/image62.png" style="width: 170.42px; height: 176.26px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                        <td class="c182" colspan="1" rowspan="1">
                            <p class="c20 c13 c7"><span class="c6"></span></p>
                            <p class="c20 c13 c7"><span class="c6"></span></p>
                        </td>
                        <td class="c110" colspan="1" rowspan="1">
                            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 191.99px; height: 177.89px;"><img alt="" src="../images/image63.png" style="width: 191.99px; height: 177.89px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13"><span class="c6">&nbsp;</span></p>
            <p class="c2 c13"><span class="c6">&nbsp;</span><span class="c27 c36">Figure 2.12</span><span class="c27 c11">: The PES of the cis-trans conversion of H</span><span class="c11 c32">2</span><span class="c27 c11">O</span><span class="c32 c11">2</span><span class="c27 c11">&nbsp;. Note that the optimum bond length is around 1.48 &Aring; and for shorter bond lengths the energy minimum is located near to 90.0o or 270.0</span>
                <span class="c11 c28">o</span><span class="c27 c11">.</span></p>
        </div>
        <div id="chapter-3">
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c76 c58 c66 c41">3</span></p>
            <p class="c79 c13 title"><span class="c29">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Modeling with Siesta</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c6">This add-on is mainly used for performing small to moderate level Siesta DFT calculations (with hundred of atoms). For larger calculations users are advised to use cluster/GCP machines with MPI parallel binaries. This add-on is equipped with OpenMP so that multi core machines can be used for its calculations, if it is found necessary. Note that Siesta 4.1 version is used in this add-on.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">At the present version all molecular structural data and non Z-matrix styled crystalline FDF data is supported by this add-on.</span></p>
            <a id="t.015e59dad276a1a227e94eb610084473cceb8465"></a>
            <a id="t.16"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c109">[1] Siesta Script Writer</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">A basic Siesta script (.fdf file) write is written; it can be used to make a template FDF file for molecular as well as crystalline systems. Most commonly used key words are added and this can be selected optionally. However user is advised to refer Siesta 4.1 manual for finalizing the script. This tool can be invoked as:</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c9"><span class="c6 c10">Set Up </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Siesta Scripting</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 376.35px; height: 214.28px;"><img alt="" src="../images/image64.png" style="width: 376.35px; height: 214.28px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 3.1</span><span class="c27 c11">: A script writer tool (FDF writer) in the Siesta add-on.</span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c42">The newly generated file is kept in: </span><span class="c55 c41">.\Input</span><span class="c6">&nbsp;directory.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">If an XYZ file is loaded into the Main Viewer, FDF file will generated with that geometry. </span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <a id="t.a2bf7d6f4eb3158ef7f4b2f270139cf19e09bfb4"></a>
            <a id="t.17"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c109">[2] Siesta Add-On and Executing the Program</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">All the basic tasks for Siesta (except its script writing) is carried out in Siesta add-on and it can be called by:</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c9"><span class="c6 c10">Execute &nbsp;</span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Siesta Calculator</span></p>
            <p class="c2 c9"><span class="c6 c10">&nbsp;</span></p>
            <p class="c2 c9"><span class="c42">It will launch the add-on as a separate program. A variety of FDF file is supported by the </span><span class="c42 c66">Jmol</span><span class="c6">&nbsp;and hence this add-on. Use &lt;Open&gt; for calling these FDF files.</span></p>
            <p class="c2 c9"><span class="c6">For example, h2o2.fdf can be opened and it can be submitted in this application, by simply going to the tab &lt;Run&gt; and clicking the Run button. The below figure shows a console output of a Siesta run. Note that the output file is stored in, </span></p>
            <p class="c2 c9"><span class="c55 c41">.\SiestaApp\SiestaBinary.4.0.CygWin64</span><span class="c42">&nbsp;directory. And its console log file is saved in .</span><span class="c4">\SiestaApp\siestaMAIN.log.</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 467.46px; height: 248.71px;"><img alt="" src="../images/image11.png" style="width: 467.46px; height: 248.71px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 3.2</span><span class="c27 c11">: Siesta log is shown in the console window of the add-on.</span></p>
            <p class="c20 c13 c7"><span class="c27 c11"></span></p>
            <p class="c2 c9"><span class="c42">In &lt;Analysis&gt; some standard or routine calculations or analysis can be done, such as checking SCF convergence, Cartesian Force components and vectors (of optimized structure), Eigenvalues etc. from the relevant files in </span>
                <span class="c55 c41">.\SiestaApp\SiestaBinary.4.0.CygWin64. </span><span class="c6">See the below figure of h2o2.fdf out file examples.</span></p>
            <a id="t.ebb99e6dbb5a23931a66768fca5c48564ce819b3"></a>
            <a id="t.18"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c42 c109">[3] Basic analayzers</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c9"><span class="c6">After running the calculations, one can use many basic analyzers, for example to check SCF convergence for single point as well as geometry optimization etc.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 438.64px; height: 180.51px;"><img alt="" src="../images/image12.png" style="width: 438.64px; height: 180.51px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 3.3</span><span class="c27 c11">: The window shows basic commands for the Siesta post processing.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c8"><span class="c6">Some of the analyzers results are shown below.</span></p>
            <ul class="c112 lst-kix_list_4-0 start">
                <li class="c2 c9 c59 li-bullet-0"><span class="c46 c41">Basic Analyzers Results: </span></li>
            </ul>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 502.61px; height: 291.11px;"><img alt="" src="../images/image13.png" style="width: 502.61px; height: 291.11px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 3.4</span><span class="c27 c11">: Results from the Siesta post processing is shown.</span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 192.00px; height: 102.15px;"><img alt="" src="../images/image14.png" style="width: 192.00px; height: 102.15px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c27 c36">Figure 3.5</span><span class="c27 c11">: </span><span class="c6">Force component vectors of each atom in H</span><span class="c19">2</span><span class="c6">O</span><span class="c19">2</span><span class="c6">&nbsp;(non equilibrium geometry).</span></p>
            <ul class="c112 lst-kix_list_4-0">
                <li class="c2 c9 c59 li-bullet-0"><span class="c46 c41">Denchar Results</span></li>
            </ul>
            <p class="c2 c13"><span class="c42">Denchar is a Siesta utility-program to plot charge densities and wave functions in real space. It can be used in 2D or in 3D. A template script for water molecule is given (but this can be modified very easily for other molecules), and its 2D/3D data can be obtained by following Steps I-IV, systematically. Note that </span>
                <span class="c69 c41">&Gamma;</span><span class="c6">&nbsp;point used in this calculations.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 455.92px; height: 175.81px;"><img alt="" src="../images/image15.png" style="width: 455.92px; height: 175.81px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c6">(a)</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 455.74px; height: 182.05px;"><img alt="" src="../images/image16.png" style="width: 455.74px; height: 182.05px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c6">(b)</span></p>
            <p class="c2 c13"><span class="c27 c36">Figure 3.6</span><span class="c27 c11">: </span><span class="c6">(a)Shows list of 2D data files of water molecule from the Denchar utility, with two different </span><span class="c0">real</span><span class="c6">&nbsp;wavefunctions (the first two low lying orbitals in a contour diagram);(b) shows equivalent 3D data.</span></p>
            <p class="c20 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c57"><span class="c6">Similary, band diagram can also be generates with template scripts.</span></p>
            <a id="t.f727949b760321cc972232d42b2d9fa1f8785d82"></a>
            <a id="t.19"></a>
            <table class="c44">
                <tbody>
                    <tr class="c143">
                        <td class="c174" colspan="1" rowspan="1">
                            <p class="c67 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 237.20px; height: 178.07px;"><img alt="" src="../images/image17.png" style="width: 237.20px; height: 178.07px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                        <td class="c175" colspan="1" rowspan="1">
                            <p class="c67 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 233.33px; height: 180.00px;"><img alt="" src="../images/image18.png" style="width: 233.33px; height: 180.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13"><span class="c36">Figure 3.7</span><span class="c27 c11">: Band and DOS diagram of Aluminium (FCC). Note that in DOS plot, there are actually &nbsp;three lines (lower orange line is composed of two lines which corresponds to spin up and down contributions, whereas green line corresponds to the total DOS).</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <a id="t.61f0b215f6c128722f3f526e65f6a61e7eb83a91"></a>
            <a id="t.20"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c73 c42">[4] Wannier Orbitals and Band Diagrams</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c42">To Wannier get localized molecular orbitals of crystalline or extended systems from the Bloch wavefunction, Wannier90.exe program is used in conjunction with the Siesta binaries. See </span><span class="c42 c66">www.wannier.org</span>
                <span class="c6">&nbsp;for more details. </span>
            </p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c2 c9"><span class="c6">In the tab &lt;Wannier DOS/Orbitals&gt; this information can be obtained by following steps I to VII (as an example template, FCC bulk Si is used). &nbsp;Results can be viewed by using &lt;Plot Bands&gt; or by &lt;View Wannier Orbitals&gt;. This template files can be easily modified for other systems.</span></p>
            <p class="c2 c9 c7"><span class="c6"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 315.11px; height: 368.53px;"><img alt="" src="../images/image2.png" style="width: 315.11px; height: 368.53px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c36">Figure 3.8</span><span class="c27 c11">: Steps of Wannier data calculations.</span></p>
            <p class="c20 c13 c7"><span class="c27 c11"></span></p>
            <p class="c2 c9"><span class="c6">&lt;Plot Bands&gt; give Wannier band diagram and &lt;View Wannier Orbitals&gt; list XSF files and that can be converted into the CUBE file format to display Wannier functions.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 495.47px; height: 210.80px;"><img alt="" src="../images/image3.png" style="width: 495.47px; height: 210.80px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c36">Figure 3.9</span><span class="c27 c11">: A Wannier orbital of bulk Si generated from a XSF file.</span></p>
            <a id="t.a35ed9953622de48f7cae115b356d5db018c54a4"></a>
            <a id="t.21"></a>
            <table class="c44">
                <tbody>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 76.67px; height: 30.67px;"><img alt="" src="../images/image1.png" style="width: 76.67px; height: 30.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
                        </td>
                    </tr>
                    <tr class="c15">
                        <td class="c53" colspan="1" rowspan="1">
                            <h3 class="c25 c13"><span class="c73 c42">[5] Molecular Vibrations</span></h3>
                        </td>
                    </tr>
                </tbody>
            </table>
            <h3 class="c75 c13 c99"><span class="c103 c42 c58"></span></h3>
            <p class="c2 c9"><span class="c6">Molecular vibrations (using &#61511; point) can be easily calculated with the application. In the tab, &lt;Vibrations&gt;, execute the steps I-V, one after another (note that example script is made for water molecule, but it can be readily modified for other molecules.). The last command, &lt;Run Vibra&gt; calculates eigenvalues/vectors and it gives an another window in which different modes can be animated. The essential molecules vibrational data is saved in .\SiestaApp\SiestaBinary.4.0.CygWin64\SiestaVibModes.xyz.</span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 495.40px; height: 216.87px;"><img alt="" src="../images/image4.png" style="width: 495.40px; height: 216.87px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c36">Figure 3.10</span><span class="c27 c11">: A vibrational frequency calculator module and its mode-displayer.</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
        </div>
        <div id="chapter-4">
            <p class="c2 c13"><span class="c76 c58 c66 c41">4</span></p>
            <p class="c79 c13 title"><span class="c66 c41">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; OpenMD Simulations </span></p>
            <p class="c94 c13 c7"><span class="c98 c36 c58"></span></p>
            <p class="c21 c13 c7"><span class="c98 c36 c58"></span></p>
            <p class="c94 c13 c7"><span class="c98 c36 c58"></span></p>
            <p class="c21 c13"><span class="c6">OpenMD add-on is aimed to perform MD calculations with OpenMD binary/scripts. This tool has four parts, (1), Viewer (2), Editor (3), Terminal and, (4), Utilities.</span></p>
            <p class="c94 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 418.00px; height: 222.67px;"><img alt="" src="../images/image5.png" style="width: 418.00px; height: 222.67px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c21 c13"><span class="c27 c36">Figure 4.1</span><span class="c61 c1 c58">: Front end of the OpenMD add-on tool displays Ag nanoparticle obtained using OpenMD.</span></p>
            <p class="c21 c13"><span class="c6">In viewer a normal XYZ file can be viewed (usually it is obtained from OMD </span><span class="c0">dump</span><span class="c6">&nbsp;file by using a converter, </span><span class="c0">Dump2XYZ</span><span class="c6">). All other standard Jmol view facilities are available here. Note that XYZ file will keep in the &lt;OpenMD/Tempview&gt; directory whereas the OMD file will be kept in the &lt;./OpenMD&gt; folder.</span></p>
            <p class="c21 c13"><span class="c6">The Editor tool will open an OMD file. Since OMD file is a kind of XML file, a special editor is used to </span><span class="c0">fold</span><span class="c6">&nbsp;and </span><span class="c0">un-fold </span><span class="c6">many sections of this file as shown in the following file.</span></p>
            <p class="c21 c13"><span class="c6">&nbsp;In Terminal-Tab, simple DOS commands can be executed. Note that one should add:</span></p>
            <p class="c21 c13"><span class="c4">cmd.exe /c</span></p>
            <p class="c21 c13"><span class="c6">before the Python scripts (eg. cmd.exe /c python.exe slabBuilder.py). The commands are stored in a stack-array and it can be easily called back (See Stored Commands).</span></p>
            <p class="c21 c13"><span class="c6">In the Utility-Tab several tools are added (eg. &lt;Clean OpenMD&gt; will delete the MD out files and &lt;Open Trajectory&gt; will open a separate viewer window to animate trajectories etc.). &nbsp;The &lt;Plot Energies&gt; will open &lt;Stat&gt; file and plot total, kinetic and potential energies. Similarly, &lt;T P V&gt; plot variation of temperature, pressure and volume during the MD run as it is stored in the &lt;Stat&gt; file.</span></p>
            <p class="c94 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 434.21px; height: 366.45px;"><img alt="" src="../images/image6.png" style="width: 434.21px; height: 366.45px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c94 c13"><span class="c27 c36">Figure 4.2</span><span class="c27 c11">: OpenMD Tool&rsquo;s XML styled editor for the OMD scripts.</span></p>
            <p class="c94 c13 c7"><span class="c6"></span></p>
            <p class="c94 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 474.00px; height: 291.00px;"><img alt="" src="../images/image7.png" style="width: 474.00px; height: 291.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c94 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 476.00px; height: 306.00px;"><img alt="" src="../images/image8.png" style="width: 476.00px; height: 306.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c94 c13"><span class="c27 c36">Figure 4.3</span><span class="c27 c11">: OpenMD Analyze variation of different energies during an MD simulation.</span></p>
            <p class="c94 c13 c7"><span class="c6"></span></p>
            <p class="c21 c13"><span class="c6">Since analyzing bond length gives an indication of reaction progress, &lt;Calculate Bond Length&gt; tool can be used to extract Bond length between two selected atoms (&ldquo;Atom Number 1&rdquo; and &ldquo;Atom Number 2&rdquo;). This will read XYZ file which </span>
                <span class="c0">doesn&rsquo;t</span><span class="c6">&nbsp;use Periodic Boundary Conditions or PBC (as an example see, SampleAnimationTest.xyz file). </span></p>
            <p class="c21 c13 c7"><span class="c6"></span></p>
            <p class="c94 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 431.00px; height: 324.00px;"><img alt="" src="../images/image9.png" style="width: 431.00px; height: 324.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c94 c13"><span class="c27 c36">Figure 4.4</span><span class="c27 c11">: An out file from the MDAnalysis script. It describes that the bond is break near 400&rsquo;th frame (in a periodic cubic box cell).</span></p>
            <p class="c94 c13 c7"><span class="c6"></span></p>
            <p class="c21 c13"><span class="c6">However, if PBC is applied more advanced script is needed to extract this bond length information. For example, if the system is equipped with Python and </span><span class="c46 c41">MDAnalysis</span><span class="c6">&nbsp;package, a user can:</span></p>
            <p class="c21 c13 c93"><span class="c6">1.</span><span class="c123 c42 c58">&nbsp; &nbsp; &nbsp; &nbsp;</span><span class="c6">Edit &lt;MDAnalysis&gt; script (</span><span class="c4">MDanalysisv1.py</span><span class="c6">) if needed, eg. to change atom numbers</span></p>
            <p class="c21 c13 c93"><span class="c6">2.</span><span class="c123 c42 c58">&nbsp; &nbsp; &nbsp; &nbsp;</span><span class="c6">Use &lt;Add MDAnalysis in Terminal&gt;, which will add cmd</span><span class="c4">.exe /c python.exe MDanalysisv1.py</span><span class="c6">&nbsp;in the Terminal window</span></p>
            <p class="c21 c13 c93"><span class="c6">3.</span><span class="c42 c58 c123">&nbsp; &nbsp; &nbsp; &nbsp;</span><span class="c6">Go to </span><span class="c46 c41">Terminal</span><span class="c6">&nbsp;and execute the above command.</span></p>
            <p class="c21 c13"><span class="c6">The sample MDAnalysis script is:</span></p>
            <a id="t.5eb7ad23294948f36fd835295d85156b3a1a4a60"></a>
            <a id="t.22"></a>
            <table class="c82">
                <tbody>
                    <tr class="c173">
                        <td class="c105" colspan="1" rowspan="1">
                            <p class="c75 c13"><span class="c26">import MDAnalysis as mda</span></p>
                            <p class="c75 c13"><span class="c26">from MDAnalysis.analysis.distances import dist</span></p>
                            <p class="c75 c13"><span class="c26">import matplotlib.pyplot as plt</span></p>
                            <p class="c75 c13"><span class="c26">import numpy as np</span></p>
                            <p class="c75 c13"><span class="c26">&nbsp;</span></p>
                            <p class="c30 c13"><span class="c26">u = mda.Universe(&#39;mdPBCTrajectoryTest.xyz&#39;)</span></p>
                            <p class="c30 c13"><span class="c26">mybox=np.array([16., 16., 16., 90., 90., 90.], dtype=&#39;f&#39;)</span></p>
                            <p class="c30 c13"><span class="c26"># 1-2&#39;th BL atoms in XYZ</span></p>
                            <p class="c30 c13"><span class="c26">distances = []</span></p>
                            <p class="c30 c13"><span class="c26">for ts in u.trajectory:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;distances.append(dist(mda.AtomGroup([u.atoms[0]]),mda.AtomGroup([u.atoms[1]]),box=[16., 16., 16., 90, 90, 90])[2][0] )</span></p>
                            <p class="c75 c13"><span class="c26">plt.switch_backend(&#39;agg&#39;)</span></p>
                            <p class="c75 c13"><span class="c26">plt.plot(distances)</span></p>
                            <p class="c75 c13"><span class="c26">plt.xlabel(&#39;Frames&#39;)</span></p>
                            <p class="c75 c13"><span class="c26">plt.ylabel(&#39;Bond Length (&Aring;)&#39;)</span></p>
                            <p class="c75 c13"><span class="c26">plt.show()</span></p>
                            <p class="c13 c75"><span class="c26">plt.savefig(&#39;1-2.png&#39;)</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c21 c13"><span class="c3">&nbsp;</span></p>
            <p class="c21 c13"><span class="c6">We strongly recommend to use MDAnalysis tools to post process the OpenMD data.</span></p>
            <p class="c21 c13"><span class="c6">The details of MDAnalysis package is available at:</span><span class="c6"><a class="c70" href="https://www.google.com/url?q=https://www.mdanalysis.org/&amp;sa=D&amp;source=editors&amp;ust=1612637412642000&amp;usg=AOvVaw0hlp048OC28P5u_SNkfTZ5">&nbsp;</a></span>
                <span class="c0">https://www.mdanalysis.org/</span>
            </p>
            <p class="c21 c13"><span class="c3">&nbsp;</span></p>
            <p class="c21 c13"><span class="c3">&nbsp;</span></p>
            <p class="c21 c13"><span class="c3">&nbsp;</span></p>
            <p class="c21 c13"><span class="c3">&nbsp;</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
        </div>
        <div id="chapter-5">
            <p class="c2 c13"><span class="c76 c58 c66 c41">5</span></p>
            <p class="c79 c13 title"><span class="c29">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Python scripting with ASE</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span>To demonstrate ASE scripting within the platform, we must create a Python project:</span></p>
            <p class="c2 c9"><span class="c6 c10">File </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;New Project </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Python </span><span class="c16 c10">&#10148;</span><span class="c6 c10">&nbsp;Python Project &ndash; Ant </span></p>
            <p class="c2 c13"><span class="c3">and save this in a suitable directory. We name the main class as &lt;ASEdemo&gt;. </span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 397.23px; height: 216.01px;"><img alt="" src="../images/image10.png" style="width: 397.23px; height: 216.01px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13"><span class="c36">Figure 5.1</span><span class="c27 c11">: Setting up a Python ASE project in the DJMol application.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 455.80px; height: 265.80px;"><img alt="" src="../images/image26.png" style="width: 455.80px; height: 265.80px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13"><span class="c36">Figure 5.2</span><span class="c27 c11">: A script window displays an ASE script.</span></p>
            <p class="c20 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span>The purpose of this ASE calculation is to obtain the </span><span class="c66">phonon dispersion for bulk aluminum</span><span class="c3">&nbsp;using a 7&times;7&times;7 supercell within effective medium theory (EMT as it is implemented in ASE). For the source, see, Ex2_phononbulkAl_DOS.py file from the example directory. After running it, this script will create the band and the DOS diagram of the system. </span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c3">The script is provided in the Example directory. Once the script is ready, use &lt;Run Project&gt; command from the &lt;Run&gt; menu to execute the script. It will initiate ASE engine and produce the band and DOS diagrams. Optionally, users can develop their own python scripts in this project directory for other tasks (eg. to save the animation of a particular mode).</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c42 c10">Run </span><span class="c10 c95 c86">&#10148;</span><span class="c42 c10">&nbsp;Run Project</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 476.70px; height: 253.31px;"><img alt="" src="../images/image31.png" style="width: 476.70px; height: 253.31px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c2 c13"><span class="c36">Figure 5.3</span><span class="c11">: (1) is the </span><span class="c1">Workplace</span><span class="c11">&nbsp;and (2) is the console output from the Python run and (3) is the resultant Band and DOS figures that is created in the </span>
                <span class="c1">Workplace</span><span class="c27 c11">.</span></p>
            <p class="c2 c13"><span class="c3">In the next advanced example (see example directory of Python) we will show that RDF (radial distribution function) of melting copper in a cubic box. Here, EMT is also used but RDF function is taking from ASAP calculator. ASAP is a calculator for doing large-scale classical molecular dynamics within the ASE package (its libraries can be compiled in CygWin or WSL). Below figure shows the RDF from ASAP calculator and uses systems terminal.</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 478.45px; height: 253.94px;"><img alt="" src="../images/image32.png" style="width: 478.45px; height: 253.94px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c30 c13"><span class="c36">Figure 5.4</span><span class="c27 c11">: Radial distribution function of melting copper (FCC) at 2500K from an ASAP calculator.</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c3">In first add Terminal into the system by:</span></p>
            <p class="c2 c13"><span class="c42 c10">&nbsp;Window </span><span class="c10 c95 c86">&#10148;</span><span class="c42 c10">&nbsp;IDE Tools </span><span class="c10 c95 c86">&#10148;</span><span class="c42 c10">&nbsp;Terminal </span></p>
            <p class="c2 c13"><span class="c3">This will get CygWin ot WSL, depends upon the settings. After this, type the code in the terminal:</span></p>
            <p class="c2 c13 c57"><span class="c27 c41 c45">python3.6m asedemo.py</span></p>
            <p class="c2 c13"><span>It will launch an ASAP calculation, and calculates RDF of the melting copper atoms. Note that its RDF is very similar to that of water and it confirms that </span><span class="c66">liquid</span><span>&nbsp;</span><span class="c66">state</span>
                <span class="c3">&nbsp;of Cu atoms at high temperature is well represented by the ASAP model.</span>
            </p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
        </div>
        <div id="chapter-6">
            <p class="c2 c13 "><span class="c76 c58 c66 c41 ">6</span></p>
            <p class="c13 c79 title "><span class="c29 ">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Tools of &nbsp;DJMol </span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <h3 class="c75 c13 c99 "><span class="c63 c58 c77 c86 "></span></h3>
            <p class="c2 c13 "><span class="c3 ">Tools are generally calculator independent (add-on) modules which can assist some steps in the modeling. </span></p>
            <p class="c2 c13 "><span class="c3 ">Some of the important Tools are described below.</span></p>
            <p class="c2 c9 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 74.47px;
            height: 30.60px; "><img alt=" " src="../images/image1.png " style="width: 74.47px; height: 30.60px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c9 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c73 c42 ">[1] Analysing Point Group</span></p>
            <p class="c2 c9 c7 "><span class="c3 "></span></p>
            <p class="c2 c9 "><span class="c6 ">DJMol use a </span><span class="c0 ">point group analyzer</span><span class="c6 ">&nbsp;and from this one can easily find the point group of a molecule and its subgroups. It can also tell the next possible point group within a specified error-bar. This module can be invoked by:</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 c10 ">Tools </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;PointGroup Analyzer</span></p>
            <p class="c2 c9 c7 "><span class="c6 c10 "></span></p>
            <p class="c2 c9 " id="h.3rdcrjn "><span class="c6 ">This tool can also be used to deform the geometry of molecules so that one can modify a molecule&rsquo;s PG into another one (in other words, we can constrain a PG into its sub group).</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 1.00px solid #d8d8d8; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 448.80px;
            height: 457.87px; "><img alt=" " src="../images/image33.png " style="width: 448.80px; height: 457.87px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 74.47px;
            height: 30.60px; "><img alt=" " src="../images/image1.png " style="width: 74.47px; height: 30.60px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 "><span class="c73 c42 ">[2] Z-Matrix Editor</span></p>
            <p class="c2 c13 "><span class="c6 ">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Z-Matrix or Z-mat tool can be called by:</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 c10 ">Extra </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Z-Matrix Tool</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">This Z-Mat editor is made for the XYZ files; It can be either used as an independent program (use File button to load a required XYZ file) or as an auxiliary program to manipulate the loaded XYZ file. Use either &lt;New File&gt; button to vary Z-matrix a of a new Cartesian file or &lt;Current File&gt; button to fetch the currently loaded geometry from the main panel of DJmol.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 438.25px;
            height: 233.28px; "><img alt=" " src="../images/image34.png " style="width: 438.25px; height: 233.28px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c20 c9 "><span class="c27 c36 ">Figure 6.1</span><span class="c27 c11 ">: A butane molecule (exaggerated) and its Z-matrix data.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c4 ">To change Z-Mat Data:</span></p>
            <p class="c2 c9 "><span class="c6 ">After displaying the geometry, select &lt;Select a Z-mat variable&gt; radio button to show the corresponding Z-matrix data (MOPAC style, generated from the OpenBabel application) of the displayed molecule in this application.</span></p>
            <p class="c2 c9 "><span class="c6 ">&nbsp;</span></p>
            <p class="c2 c9 "><span class="c6 ">Also select any single cell from either &lt;X&gt;, or &lt;Y&gt; or &lt;Z&gt; column, which represents bond-length, bond angles, or dihedral angles, respectively. After that select &lt;Insert Variable&gt; button to make the cell element as a Z-mat variable. </span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">And then use the &lt;Save&gt; button and close the window. After that, immediately select the &lt;Start Zmatrix Variation&gt; menu from the top of the window, and by using [+] and [-] buttons, start the variation. </span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">One can either save the data (&lt;Update Zmat&gt; button) or discard the variation and to move the molecule into its original state by using &lt;Undo Update&gt; button.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">It is also important to note that to stop and undo the current variation,first to select &lt;Stop Variations&gt; and press &lt;Undo Update&gt; button&gt;.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">If you want to move only X or Y or Z positions of a single atom use the menu ( &lt;Move selected Atom&gt; ), and select an atom (use a click on the desired atom) , and press any[x+][x-]...buttons near the bottom left corner.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c4 ">Important</span><span class="c6 ">: </span><span class="c0 ">To save the updated Geometry press &lt;Update Zmat&gt; </span><span class="c46 c41 ">AND then </span><span class="c0 ">&lt;Save Structure&gt; button.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">Or to discard the variation and to move the molecule into its original state by using &lt;Undo Update&gt; button.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">Also don&#39;t forget to use &lt;Save Structure&gt; button to save the modified Cartesian file. It will be saved as </span><span class="c4 ">CurrentFinalXYZ</span><span class="c6
            ">.</span><span class="c4 ">xyz</span><span class="c6 ">&nbsp;file in the folder &lt;</span>
                <span class="c4 ">ZMATfolder</span><span class="c6 ">&gt; in the program directory.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">If necessary use:</span></p>
            <p class="c2 c9 "><span class="c6 ">[ 1 ] &lt;Select Atoms Labels&gt; &nbsp;to see individual atom labels, which may be useful when one manipulates Z-Mat.</span></p>
            <p class="c2 c9 "><span class="c6 ">[ 2 ] &lt;Show initial bonds&gt; for keeping the original bonding pattern, irrespective of the current geometric state. This can be useful when some bonds are too far away from the initial bonded state.</span></p>
            <p class="c2 c9 "><span class="c6 ">[ 3 ] &lt;Show axis&gt; to show cartesian axis</span></p>
            <p class="c2 c9 "><span class="c6 ">[ 4 ] &lt;Add New Bond&gt;[n1] [n2] - to show a new bond between the atomlabel, n1 to atomlabel2, n2 atoms.</span></p>
            <p class="c2 c9 "><span class="c6 ">[ 5 ] &lt;Builder&gt; option can be used if necessary.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">Note that a user can also enter numerical values (in float) to the Text Fields. Also Move the slide bar to maximum when you vary the Angles/Dihedral angles and to a minimum if you want to manipulate the distance.</span></p>
            <p class="c2 c13 c7 " id="h.26in1rg "><span class="c103 c42 c58 "></span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 74.47px;
            height: 30.60px; "><img alt=" " src="../images/image1.png " style="width: 74.47px; height: 30.60px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 "><span class="c73 c42 ">[3] Remote Submission of Scripts</span></p>
            <p class="c2 c9 "><span class="c6 ">For time consuming job, such as MD calculations, it is a better idea to use a dedicated machine like Linux cluster/Google Cloud etc. instead of a PC. In this case, one need to generate SSH keys (ie. one private and one public key), preferably, by using PUTTYGEN application; see </span>
                <span class="c42 c97 "><a class="c70 " href="https://www.google.com/url?q=https://www.puttygen.com/&amp;sa=D&amp;source=editors&amp;ust=1612637412654000&amp;usg=AOvVaw2tDz5mRORzldWUSJN9Vq-k
            ">https://www.puttygen.com/</a></span><span class="c6 ">&nbsp;to download this mandate application.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">Note that we need &lt;RSA&gt; based private key (which is stored in your PC) and a public key (stored in the Remote machine). Note that you must enter a &lt;Passphrase&gt; for the &lt;Private key&gt; and its format should be in the </span>
                <span class="c4 ">OpenSSH </span><span class="c6 ">(ie. not in the PPK format, which is the PuttyGens&#39; default). PuttyGEN can convert this private key in PPK format to OpenSSH format. Finally the Private Key should be kept in the &lt;ssh&gt; folder with a file name, &lt;id_rsa&gt;. Make sure that the file &lt;known_hosts&gt; is existed in &lt;ssh&gt; format. After this one can invoke the &lt;Remote Submission Tool&gt; as:</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 c57 "><span class="c6 c10 ">RunProject </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Remote Submission</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c9 "><span class="c6 ">This tool can be used instead of PUTTY, and it is equipped with tools for file upload/download purposes. The downloaded files are kept in &lt;SCP&gt; folder, by default.</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c20 c13 image-container"><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px);
            -webkit-transform: rotate(0.00rad) translateZ(0px); width: 431.29px; height: 393.90px; "><img alt=" " src="../images/image35.png " style="width: 431.29px; height: 393.90px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px);
            -webkit-transform: rotate(0.00rad) translateZ(0px); " title=" "></span></p>
            <p class="c154 c13 " id="h.lnxbz9 "><span class="c27 c36 ">Figure 6.2</span><span class="c27 c11 ">: A sample SSH session (for Google Cloud Platform remote connection and file transactions).</span></p>
            <p class="c154 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c94 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 74.47px;
            height: 30.60px; "><img alt=" " src="../images/image1.png " style="width: 74.47px; height: 30.60px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <h3 class="c75 c13 "><span class="c73 c42 ">[4] Version Control using GIT</span></h3>
            <p class="c21 c13 "><span class="c6 ">Version control (VC) systems are a category of software tools that help a software team manage changes to source code over time. Version control software keeps track of every modification to the code in a special kind of database. If a mistake is made, developers can turn back the clock and compare earlier versions of the code to help fix the mistake while minimizing disruption to all team members. </span>
                <span class="c4 ">Git </span><span class="c6 ">is an example of VCS.</span></p>
            <p class="c21 c9 "><span class="c6 ">Why the VC is used in the DJMol? Since the text based input scripts are playing an important role in the modeling (inputs of Siesta/DFTB/ASE, all are text files). Using VC, these data can be re-edited remotely and at the same time it keeps different version of these files in a systematic way (for example, later, it can be used by collaborators/public).</span></p>
            <p class="c21 c9 "><span class="c6 ">Some Basic Definitions to be familiarized are:</span></p>
            <p class="c21 c9 "><span class="c6 ">1. &nbsp;</span><span class="c4 ">FETCH: </span><span class="c6 ">The git fetch command downloads commits, files, and refs from a remote repository into your local repo. Fetching is what you do when you want to see what everybody else has been working on.</span></p>
            <p class="c21 c9 "><span class="c6 ">2. &nbsp;</span><span class="c4 ">PULL: </span><span class="c6 ">The git pull command is used to fetch and download content from a remote repository and immediately update the local repository to match that content.</span></p>
            <p class="c21 c9 "><span class="c6 ">3. &nbsp;</span><span class="c4 ">PUSH: </span><span class="c6 ">The git push command is used to upload local repository content to a&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;remote repository. Pushing is how you transfer commits from your local repository to a remote repo. It is the counterparts to git fetch.</span></p>
            <p class="c9 c88 "><span class="c46 c41 ">How to use Git to push and pull in DJMol?</span></p>
            <p class="c21 c13 c111 "><span class="c6 ">1. &nbsp;Initializing Version Control.</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">a. &nbsp;Set up a Git Remote Repository in GitHub and copy the Repository URL.</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">b. &nbsp;Open the project in which you need to use version Control.</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">c. &nbsp;Initializing the Local Repository.</span></p>
            <p class="c21 c8 c47 "><span class="c6 c10 ">Team </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Git </span><span class="c10 c16 ">&#10148;</span><span class="c6 c10 ">&nbsp;Initialize </span><span class="c16 c10
            ">&#10148;</span><span class="c6 c10 ">&nbsp;(OK)</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">d. &nbsp;Linking the Local Repository and the online Repository</span></p>
            <p class="c21 c13 c78 c57 "><span class="c6 c10 ">Team </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Remote </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Pull </span><span class="c16
            c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Fill </span>
                <span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Check Master </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Finish</span></p>
            <p class="c21 c13 c120 "><span class="c6 c10 ">Specify &nbsp;</span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;PateTheRepository </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10
            ">&nbsp;UserName/PassWord (if needed) </span><span class="c16 c10 ">&#10148;</span>
                <span class="c6 c10 ">&nbsp;Next</span>
            </p>
            <p class="c21 c13 "><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2. &nbsp;Push</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">a. &nbsp;After making necessary changes in the Project.</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">b. &nbsp;From the Project sidebar, add the needed files to the repository by right-clicking on the file and Clicking on </span><span class="c4 ">Add</span><span class="c6 ">.</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">c. &nbsp;Commit the files to the Repository</span></p>
            <p class="c21 c8 c47 "><span class="c6 c10 ">Team </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Commit </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10
            ">&nbsp;AddACommitMessage </span><span class="c16 c10 ">&#10148;</span>
                <span class="c6 c10 ">&nbsp;Commit</span>
            </p>
            <p class="c21 c13 c47 "><span class="c6 ">d. &nbsp;Push to remote repository.</span></p>
            <p class="c21 c13 c57 c78 "><span class="c6 c10 ">Team </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Remote </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10
            ">&nbsp;PushToUpstream </span><span class="c16 c10 ">&#10148;</span>
                <span class="c6 c10 ">&nbsp;Yes</span>
            </p>
            <p class="c21 c13 c57 "><span class="c6 ">3. &nbsp;Pull</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">a. &nbsp;Fetch</span></p>
            <p class="c21 c13 c78 c57 "><span class="c6 c10 ">Team </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Remote </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Fetch </span><span class="c16
            c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Next </span>
                <span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Finish</span></p>
            <p class="c21 c13 c47 "><span class="c6 ">b. &nbsp;Pull</span></p>
            <p class="c21 c13 c78 c57 "><span class="c6 c10 ">Team </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Remote </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Pull </span><span class="c16
            c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Next </span>
                <span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;CheckMaster </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Finish</span></p>
            <p class="c2 c9 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c4 ">Installation of Git</span></p>
            <p class="c21 c13 "><span class="c6 ">Adding Github Plugins</span></p>
            <p class="c21 c13 "><span class="c6 ">After installing the application, search &lt;Git&gt; in the plug-ins search field. Select the &lt;Git&gt; from the &lt;Installed Packages&gt; and click on &lt;Activate&gt; as shown below figure.</span></p>
            <p class="c21 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 600.00px;
            height: 395.00px; "><img alt=" " src="../images/image30.png " style="width: 600.00px; height: 395.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c20 c13 "><span class="c27 c36 ">Figure 6.3</span><span class="c27 c11 ">: Installation of Github plugin in DJMol.</span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 74.47px;
            height: 30.60px; "><img alt=" " src="../images/image1.png " style="width: 74.47px; height: 30.60px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <h3 class="c75 c13 "><span class="c73 c42 ">[5] Database File Retrieving</span></h3>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">To fetch different structural file form various open file repositories, a Database tool is implemented. It is essentially uses Jmol&rsquo;s DB module.</span></p>
            <p class="c2 c13 "><span class="c6 ">It can be launched by:</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c21 c13 c78 c57 "><span class="c6 c10 ">Extra </span><span class="c16 c10 ">&#10148;</span><span class="c6 c10 ">&nbsp;Database Tool </span></p>
            <p class="c2 c13 "><span class="c0 ">Open URL</span><span class="c6 ">: It can be used to open XYZ or PDB file or any other Jmol recognizable file from the web server, say from Github. &nbsp;</span></p>
            <p class="c2 c13 "><span class="c6 ">Open MOL: It is mainly for, SMILES, InChI, or CAS from either a PubChem database or from NCI/NIH database. </span></p>
            <p class="c2 c13 "><span class="c6 ">Open PDB: It use RCSB web (please prefer RCSB) to load 4-character PDB ID (eg. 1crn) or 3 -character ligand (eg. 60C). </span></p>
            <p class="c2 c13 "><span class="c6 ">Open COD: It opens a specific COD ID from </span><span class="c0 ">http://www.crystallography.net/cod/</span><span class="c6 ">&nbsp;</span></p>
            <p class="c2 c13 "><span class="c6 ">Open Materials Project: It opens a specific Materials Project ID number: </span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">All opened file is saved in ./Database folder from &nbsp; </span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Export To: Only MOL, XYZ and PDB formats are supported for export its images.</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 495.13px;
            height: 313.27px; "><img alt=" " src="../images/image21.png " style="width: 495.13px; height: 313.27px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c20 c13 "><span class="c27 c36 ">Figure 6.4:</span><span class="c27 c11 ">&nbsp;Showing a Database window with a retrieved COD file from COD databse.</span></p>
            <p class="c30 c13 c7 "><span class="c27 c11 "></span></p>
            <h3 class="c75 c13 "><span class="c73 c42 ">[6] Miscellaneous Tools</span></h3>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c30 c13 "><span class="c3 ">Some of the miscellaneous tools are shown below:</span></p>
            <p class="c2 c9 "><span class="c55 c41 ">Start Batch Process: </span><span class="c42 ">For DFTB+ calculation, a number of HSD files in a particular folder can be called one after another (known as </span><span class="c42 c66
            ">batch</span><span class="c42 ">&nbsp;</span>
                <span class="c42 c66 ">processing</span><span class="c6 ">). This will be useful for building 1D PES or to analyze energies of particular set of molecules. The resultant TGZ files contain all the out files including the submitted HSD file, and it will be transformed to the folder at the end of the each calculation.</span></p>
            <p class="c20 c9 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 495.60px;
            height: 379.87px; "><img alt=" " src="../images/image22.png " style="width: 495.60px; height: 379.87px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c9 "><span class="c55 c41 ">Calculator: </span><span class="c6 ">System&rsquo;s default calculator can be invoked.</span></p>
            <p class="c2 c9 "><span class="c55 c41 ">GIF Utility: </span><span class="c42
            ">GIF animation files can be viewed or generated using this utility (eg. phonon modes). Time between two frames can be adjusted and the resultant file is stored in </span><span class="c4 ">./Scratch_images.</span></p>
            <p class="c2 c9 "><span class="c55 c41 ">Process Status: </span><span class="c42 ">It is used to monitor current resources of the computer system (used/availble RAM, HDD space etc.). This can be used before starting a calculation. For example, if the PC memory is low, allocate more RAM by stopping other less important process by using Window&rsquo;s </span>
                <span class="c55 c41 ">Task Manager</span><span class="c6 ">&nbsp;utility.</span></p>
            <p class="c2 c9 "><span class="c55 c41 ">Unit Conversion: </span><span class="c6 ">A basic unit converter for length, time, energy etc. it also includes atomic unit.</span></p>
            <p class="c2 c9 "><span class="c55 c41 ">Matrix Viewer: </span><span class="c42 ">2D matrix data viewer of the program. It is useful to analyze the overall shape and symmetry of the matrix data, such as Hessian or Hamiltonian. See example directory for its sample file.</span></p>
            <p class="c2 c9 "><span class="c55 c41 ">Convert Structure: </span><span class="c42 c66 ">Open</span><span class="c42 ">&nbsp;</span><span class="c42 c66 ">Babel</span><span class="c6 ">&nbsp;is used to convert from one structure data into another data. See Figure 2.10.</span></p>
            <p class="c2 c9 "><span class="c55 c41 ">Standard Orientation: </span><span class="c42 ">Using symmetry a disoriented molecule can be oriented with respect to a symmetry axis. This tool is useful for making a more systematic Z-matrix or Cartesian file</span>
                <span class="c42 c66 ">. It is strongly recommended that the standard oriented geometry data should be used in the Z-matrix tool to re-adjust its coordinates.</span>
                <span class="c6 ">&nbsp;</span>
            </p>
            <p class="c20 c9 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 494.80px;
            height: 265.00px; "><img alt=" " src="../images/image23.png " style="width: 494.80px; height: 265.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c30 c9 c7 "><span class="c3 "></span></p>
            <p class="c2 c9 "><span class="c55 c41 ">Open DJMol Directory</span><span class="c6 ">: It will open the parent directory. </span></p>
            <p class="c2 c9 "><span class="c41 c55 ">Send Message</span><span class="c6 ">: User can mail the forum using this utility. Please academic E-mail ID (avoid .com E-mail IDs &ndash; it won&rsquo;t support). If needed images should be linked as an URL link (say, by drive.google.com/&hellip; link).</span></p>
            <p class="c30 c9 c7 "><span class="c3 "></span></p>
            <p class="c30 c9 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c76 c58 c66 c41 "></span></p>
        </div>
        <div id="chapter-7">
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c76 c58 c66 c41">7</span></p>
            <p class="c79 c13 title"><span class="c29">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Architecture of the Program</span></p>
            <p class="c30 c9 c7"><span class="c3"></span></p>
            <h4 class="c92 c13"><span class="c63 c58 c77 c86"></span></h4>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c2 c13"><span class="c80 c58 c55 c41">Software Architecture</span></p>
            <p class="c2 c13 c7"><span class="c6"></span></p>
            <p class="c2 c13"><span class="c6">In practice, computational materials science research encompasses three distinctive steps, viz. (1), constructing the geometrical data or structure of the system (2), performing computations and, (3), obtaining results by the post processing of calculated data. All these steps can be effectively coupled by means of modeling platform such that a user can create, visualize, and share various input data files and run it with appropriate programs to obtain output data to analyze the results.</span></p>
            <p class="c2 c13"><span class="c6">By integrating molecular (or crystal) visualizer, scripting tools (in this case, Python and Jmol scripting) and other standard features of an IDE (integrated development environment), a user can interactively build or manipulate structures of materials or molecules, and its atomistic properties can be calculated with appropriate ab initio programs from either a local or a remote machine. A number of built-in tools, scripts are provided for the analysis purpose. To demonstrate the platform we have used DFTB+ as well as Siesta electronic structure code along with ASE (atomic simulation environment) and OpenMD molecular dynamics package.</span></p>
            <p class="c2 c13"><span class="c6">&nbsp; &nbsp;The programming language, Java (version 8) was chosen for this project since it is free, strongly object-oriented and shows its neutrality to the various operating systems. The object oriented programming is arguably the most popular software development technique which effectively manages the complexity in code development [9]. We preferred Java, since (1) it does not use pointers, and (2), it supports threads implicitly. Although pointers provide direct access to the computer memory, careless use of the pointers will easily leads to segmentation fault and other vulnerables. It is our personal opinion that the bugs emerged from the improper application of pointers are difficult to trace or debug, in addition to this, code with pointers is difficult to translate into other computer languages without the pointer features. Concerning threads, in this application, we used a number of threads (i.e. user threads), apart from the JVM generated daemon threads. Since the threads are inherently supported by the language one can readily create a thread by extending a Thread class. Exception handling of Java is equipped with two types, namely, unchecked and checked exception handling. And in most of the cases we used checked exceptions, as it is less tedious to implement.</span></p>
            <p class="c2 c13"><span class="c42">&nbsp; &nbsp;The core of the DJMol application consists of (Apache-) Netbeans Platform and it can be regarded as the engine behind the Netbeans IDE. In other words, many of the technical features of DJMol program is </span>
                <span class="c42 c66">inherited</span><span class="c42">&nbsp;from Netbeans IDE which is initially designed for developing Java/Javascript and C/C++ applications and consists a number of utilities to increase the productivity of a user (for example, it has an advanced sourcecode editor with code completion utilities, tools for refactoring, version control systems, Git based collaboration tools etc.) and these utilities can also be used efficiently for various modeling or scripting tasks. Moreover, this platform consists of a set of independent modular software components called modules (e.g., an SSH module to communicate with an external cloud platform). Apart from this, the program supports plug-ins so that one can selectively add or remove new features into it (e.g. Python interpreter) without being re-compiled the code. Optionally, users of the code can make their own plug-ins, for example, to incorporate another </span>
                <span class="c42 c66">ab initio</span><span class="c6">&nbsp;package. See the below for the schematic structure of the program. To the best of our knowledge, DJMol is the first opensource modeling platform which is built from a programming IDE.</span></p>
            <p class="c2 c13"><span class="c42">To display the molecular, crystal, nanostructures a Java library, Jmol, has been embedded into the program. The Jmol program is an open source, cross-platform and a highly independent (i.e. it does not depend on any third party libraries like Java3D or OpenGL) 3D visualization application. Apart from this, there are two distinct features associated with Jmol: (1) Jmol can be programmatically embedded into any Java code which uses Swing application programming interface (2), It supports an internal command scripting, so that a user can control or manipulate the display or send and retrieve parameters, data, commands etc. For example, to manipulate Gaussian cube files a user can effectively apply this internal scripting ability of Jmol. Unlike most of the visualizers, Jmol is capable of perceiving the molecular structure in three dimensions by applying stereographic projections. A viewer, with appropriate anaglyph spectacles, 3D perspective images of the molecules can be interactively visualized. Apart from this, Jmol consists of an UFF (uniform force field) method and it can be used to pre-optimize a variety of organic as well as inorganic molecules. To create 2D data plots we have used either Matplotlib based scripts or Jfreechart libraries. In the case of 3D data plots, (for example, the potential energy surfaces or orbital contour diagrams), a Java based opensource library </span>
                <span class="c42 c66">jzy3d</span><span class="c6">&nbsp;has been used.</span></p>
            <p class="c2 c13 c7"><span class="c3"></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 288.18px; height: 189.60px;"><img alt="" src="../images/image24.png" style="width: 288.18px; height: 189.60px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c56 c13"><span class="c36">Figure 7.1</span><span class="c27 c11">: A schematic diagram of DJMol software architecture. The core of the program is an IDE which is connected to other modules or programs.</span></p>
            <p class="c56 c13 c7"><span class="c27 c86 c102"></span></p>
            <p class="c56 c13 c7"><span class="c27 c86 c102"></span></p>
            <p class="c56 c13 c7"><span class="c27 c86 c102"></span></p>
            <p class="c13 c127 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 323.20px; height: 197.43px;"><img alt="" src="../images/image25.png" style="width: 323.20px; height: 197.43px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c71 c13"><span class="c36">Figure 7.2</span><span class="c27 c11">: A standard workflow for a modeling task under the DJMol program (mandate workflows are indicated by black arrows). </span></p>
            <p class="c71 c13"><span class="c11">Java&rsquo;s native thread is used to initiate add-ons so that its console can be controlled independently, if needed. Below figure shows Windos&rsquo; </span><span class="c36">Task managers</span><span class="c27 c11">&nbsp;snapshot of OpenMD add-on, while it is under running (and it indicate two independent processes). </span></p>
            <p class="c71 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 495.67px; height: 216.47px;"><img alt="" src="../images/image27.png" style="width: 495.67px; height: 216.47px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c11 c27"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c13 c7 c71"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
            <p class="c71 c13 c7"><span class="c27 c11"></span></p>
        </div>
        <div id="chapter-8">
            <p class="c71 c13 "><span class="c108 c66 c41 ">8</span></p>
            <p class="c79 c13 title "><span class="c29 ">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; software Applications</span></p>
            <p class="c9 c7 c30 "><span class="c3 "></span></p>
            <h4 class="c92 c13 "><span class="c63 c58 c77 c86 "></span></h4>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c80 c77 c41 c109 "></span></p>
            <p class="c2 c13 "><span class="c80 c77 c109 c41 ">Demonstration of DJMol - 1 </span></p>
            <p class="c2 c13 "><span class="c58 c77 c41 c80 ">UV-Visible Spectrum of Silver Nanocluster using DFTB+</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c42 ">To demonstrate the basic functionality of the software, here we describe how an UV-Visible spectrum of a silver nanocluster can be calculated using DFTB+ code in conjunction with the DJMol software. There are three steps to calculate the spectrum, </span>
                <span class="c42 c66 ">viz. </span><span class="c42 ">(1) Obtain the structure of Ag nanocluster, (2) Optimize the structure and create the corresponding HSD file for the TD-DFTB calculation and, (3) Plot the convoluted Spectrum data. Note that we arbitrarily choose Ag</span>
                <span class="c42 c62 ">177</span><span class="c42 ">&nbsp;nanocluster (its diameter is around 1.5 nm) for the TD-DFTB calculation. And the geometry of this cluster is generated by OpenMD package &nbsp;(in CygWin environment) by using experimental lattice</span><span>&nbsp;</span>
                <span class="c42 ">constant of silver (4.08 A) and FCC lattice structure. The following commands are used in the </span><span class="c42 c66 ">DJMol</span><span class="c42 ">&nbsp;</span><span class="c42 c66
            ">terminal</span><span class="c42 ">&nbsp;to obtain the Cartesian coordinate data of the system:</span></p>
            <a id="t.3512244646fbcb4fb835432746a0f92d66bc3dcb "></a>
            <a id="t.23 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c38 " colspan="1 " rowspan="1 ">
                            <p class="c115 c13 c7 "><span class="c3 "></span></p>
                            <p class="c115 c13 "><span class="c26 ">cmd.exe /c &nbsp;nanoparticleBuilder --latticeConstant=4.08 --radius=15 Ag.omd -o NP15.omd</span></p>
                            <p class="c13 c115 "><span class="c34 c113 ">cmd.exe /c &nbsp;Dump2XYZ -i NP15.omd</span><span class="c34 c58 c91 ">&nbsp;</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c115 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c42 ">The resultant coordinate data (XYZ file), can be visualized by DJMol program to generate a </span><span class="c42 c66 ">dftb_in.hsd</span><span class="c42 ">&nbsp;file using the </span><span class="c42
            c66 ">DFTB+ Script Writer </span>
                <span class="c42 ">tool with appropriate settings, for example, by adding a conjugate gradient method. See </span><span class="c42 c66 ">SetUp</span><span class="c42 ">&nbsp;menu for this utility. This HSD file can be then used to generate an optimized Ag</span>
                <span class="c42 c62 ">177</span><span class="c42 ">&nbsp;cluster by executing a DFTB+ calculation. It can be seen that this optimized cluster (See </span><span class="c42 c66 ">geo_end.xyz</span><span class="c42 ">&nbsp;file for the data) is approximately spherical in symmetry. The optimized structure was again visualized and used in the </span>
                <span class="c42 c66 ">Script Writer</span><span class="c6 ">&nbsp;tool to make another HSD file to include instructions for Casida method (a TD-DFTB algorithm). &nbsp;The essential part of this script is:</span></p>
            <a id="t.7d0320695518b208958c8d15a64a2b41da35e0f1 "></a>
            <a id="t.24 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c38 " colspan="1 " rowspan="1 ">
                            <p class="c119 c13 "><span class="c52 ">. . .</span></p>
                            <p class="c119 c13 "><span class="c26 ">ExcitedState = Casida {</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; NrOfExcitations=100</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; StateOfInterest=0</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; Symmetry=singlet</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; WriteTransitions=no</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; WriteSPTransitions=yes</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; WriteMulliken=no</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; WriteCoefficients=no</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; WriteEigenvectors=no</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; WriteTransitionDipole=no</span></p>
                            <p class="c119 c13 "><span class="c26 ">&nbsp; &nbsp; OscillatorWindow=0.01</span></p>
                            <p class="c13 c119 "><span class="c26 ">}</span></p>
                            <p class="c119 c13 "><span class="c113 ">. . .</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c115 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c42 ">In all the DFTB calculation, we used </span><span class="c42 c66 ">hyb-0-2</span><span class="c42 ">&nbsp;parameter set is used for the silver atoms. This newly generated HSD file is submitted to a remote Linux machine using the SSH utility (See </span>
                <span class="c42 c66 ">Extra</span><span class="c42 ">&nbsp;menu in the tool bar) of the DJMol and after finishing the calculation a SPX output file was retrieved in the local machine, which contains the oscillator strengths of different electronic excitations. It was then plotted with </span>
                <span class="c42 c66 ">UV-Vis spectrum</span><span class="c42 ">&nbsp;from the </span><span class="c42 c66 ">Tool</span><span class="c42 ">&nbsp;menu to obtain the spectrum of the molecule. A Lorentzian convolution (with 0.035 eV of full width at half maximum) is used to obtain the spectrum as shown in the Fig.X. Note that the </span>
                <span class="c42 c66 ">absorption</span><span class="c42 ">&nbsp;</span><span class="c42 c66 ">maximum</span><span class="c42 ">&nbsp;wavelength (</span><span class="c85 ">&lambda;</span><span class="c42 c62 ">max</span><span class="c42 ">) is located around 440 nm </span>
                <span class="c42 c66 ">i.</span><span class="c42 ">e. around the indigo-blue region. By plotting molecular orbital energy levels (using </span><span class="c42 c66 ">MOLevels</span><span class="c42
            ">&nbsp;tool and with the </span><span class="c42 c66 ">detailed.out </span>
                <span class="c42 ">file) one can seen that the Fermi level is near to 3.9 eV and band gap of this cluster is negligibly small (See the inset figure in Fig. X). And it qualitatively indicates that Ag</span><span class="c42 c62 ">177</span>
                <span class="c6 ">&nbsp;behaves like a semi-metal.</span>
            </p>
            <p class="c2 c13 "><span class="c42 ">Different sized nanoparticles can be constructed by applying the above said method to study </span><span class="c42 c66 ">red</span><span class="c42 ">&nbsp;</span><span class="c42 c66
            ">shift</span><span class="c42 ">&nbsp;property of the system. See the Fig.XX. Experimentally this is a known fact for the Ag</span>
                <span class="c42 c62 ">n </span><span class="c42 ">system (By comparing </span><span class="c85 ">&lambda;</span><span class="c42 c62 ">max</span><span class="c42 ">&nbsp;values of clusters, we can seen that, </span><span class="c85 ">&lambda;</span>
                <span class="c42 c62 ">max</span>
                <span class="c42 ">&nbsp;shifts towards UV region if we increase the size of the nano clusters.).</span><span class="c42 c177 ">&nbsp;</span><span class="c42 ">This can be qualitatively justified by </span><span class="c42 c66 ">particle in a box</span>
                <span class="c42 ">&nbsp;analogy. Note that the diameter of Ag</span>
                <span class="c42 c62 ">177</span><span class="c42 ">&nbsp;is longer than that of the Ag</span><span class="c42 c62 ">13</span><span class="c42 ">&nbsp;so that its electron can move longer along its diameter (the </span><span class="c42
            c66 ">box</span>
                <span class="c42 ">&nbsp;</span>
                <span class="c42 c66 ">length</span><span class="c42 ">). This will reduce its </span><span class="c85 ">&Delta;</span><span class="c42 ">E values significantly (by increasing the size of the diameter of a silver cluster, a photon from the </span>
                <span class="c42 c66 ">red</span>
                <span class="c6 ">&nbsp;region - a wavelength that corresponds to a lower energy - can cause an electronic excitation). For experimental UV-Vis spectra, see: https://www.sigmaaldrich.com/technical-documents/articles/materials-science/nanomaterials/silver-nanoparticles.html.</span>
            </p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c13 c131 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 481.35px;
            height: 256.45px; "><img alt=" " src="../images/image28.png " style="width: 481.35px; height: 256.45px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c39 c13 "><span class="c36 ">Figure 8.1:</span><span class="c11 ">&nbsp;The UV-Visible spectrum of Ag</span><span class="c11 c62 ">177</span><span class="c11 ">&nbsp;nanocluster as it is plotted with the </span><span class="c1 ">UV-Vis Spectrum</span>
                <span class="c27 c11 ">&nbsp;tool. The vertical red line shows the oscillator strengths of electronic transitions. The inset figure shows HOMOs (green lines) and LUMOs (red lines) of the system.</span>
            </p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c13 c7 c56 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c151 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 266.40px;
            height: 405.64px; "><img alt="All Ag nano UVVS.png " src="../images/image29.png " style="width: 266.40px; height: 405.64px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c13 c151 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 386.46px;
            height: 174.45px; "><img alt=" " src="../images/image19.png " style="width: 386.46px; height: 174.45px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 "><span class="c36 ">Figure 8.2</span><span class="c11 ">: Sizes of different AG-nanoparticle and its UV-Vis spectrum. Note the </span><span class="c1 ">red shift</span><span class="c27 c11 ">&nbsp;of </span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c56 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c2 c13 "><span class="c80 c77 c109 c41 ">Demonstration of DJMol - 2</span></p>
            <p class="c2 c13 "><span class="c117 c41 ">Frontier Orbitals of C</span><span class="c62 c41 c117 ">60 </span><span class="c80 c58 c77 c41 ">Calculated with Siesta</span></p>
            <p class="c2 c13 "><span>Volumetric MO data can be made by using executing steps described in </span><span class="c41 ">Denchar Utility</span><span class="c3 ">, systematically using the Siesta add-on. The Siesta add-on can be invoked by: &nbsp;</span></p>
            <p class="c20 c13 "><span class="c42 c10 ">Execute </span><span class="c10 c95 c86 ">&#10148;</span><span class="c42 c10 ">&nbsp;Siesta Calculator</span></p>
            <p class="c2 c13 "><span>In this section we shall obtain the </span><span class="c66 ">frontier</span><span>&nbsp;</span><span class="c66 ">orbitals</span><span>&nbsp;of Buckminster fullerene. See the example file </span><span class="c66 ">Fullerene.fdf</span><span>&nbsp;under the Siesta_Samples folder for its coordinate (note that it is an un-optimized structure; its optimization procedure is left to the user as an additional exercise). &nbsp;Since we used a molecular system in a unit cell, the default k-point (</span>
                <span class="c66 c41 c69 ">&Gamma;</span><span>-point) is used to represent electronic wavefunction coefficients. Note that HOMO and LUMO coefficients are at 120 and 121&rsquo;th </span><span class="c66 c41 ">k</span><span class="c3 ">-points. See the Siesta Manual for more details.</span></p>
            <p class="c2 c13 "><span>And change the main setting of the fdf file as (ie. except the coordinate as it is inserted between, </span><span class="c66 ">AtomicCoordinatesAndAtomSpecies</span><span class="c3 ">&nbsp;block):</span></p>
            <a id="t.311a4371bfa667dcf618ce9c927c2ea168b917a2 "></a>
            <a id="t.25 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c33 c13 "><span class="c12 ">#------------------------------------------------</span></p>
                            <p class="c33 c13 "><span class="c12 "># &nbsp; IMPORTANT: DO NOT CHANGE SystemLabel</span></p>
                            <p class="c33 c13 "><span class="c12 ">#------------------------------------------------</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">SystemName &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;Fullerene molecule &nbsp; &nbsp; &nbsp; &nbsp;# Descriptive name of the system </span></p>
                            <p class="c33 c13 "><span class="c12 ">SystemLabel &nbsp; &nbsp; &nbsp; &nbsp; siesta &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# Short name for naming files</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">NumberOfAtoms &nbsp; &nbsp; &nbsp; 60</span></p>
                            <p class="c33 c13 "><span class="c12 ">NumberOfSpecies &nbsp; &nbsp; 1</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">SpinPolarized &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; T</span></p>
                            <p class="c33 c13 c7 "><span class="c123 c34 c109 "></span></p>
                            <p class="c33 c13 "><span class="c51 c41 c45 ">WriteDenchar &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;T</span></p>
                            <p class="c33 c13 "><span class="c51 c41 c45 ">WriteWaveFunctions &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;T</span></p>
                            <p class="c33 c13 "><span class="c51 c41 c45 ">COOP.Write &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;T</span></p>
                            <p class="c33 c13 c7 "><span class="c123 c34 c109 "></span></p>
                            <p class="c33 c13 "><span class="c51 c41 c45 ">%block WaveFuncKpoints</span></p>
                            <p class="c33 c13 "><span class="c51 c41 c45 ">&nbsp; 0.000 0.000 0.000 from 120 to 121 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# Gamma wavefuncs Here, 120 = HOMO </span></p>
                            <p class="c33 c13 "><span class="c51 c41 c45 ">%endblock WaveFuncKPoints</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">%block ChemicalSpeciesLabel</span></p>
                            <p class="c33 c13 "><span class="c12 ">&nbsp;1 &nbsp;6 &nbsp;C &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; # SpeciesIndex,AtomicNumber,SpeciesLabel</span></p>
                            <p class="c33 c13 "><span class="c12 ">%endblock ChemicalSpeciesLabel</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">LatticeConstant &nbsp; &nbsp; &nbsp; 25.0 Ang</span></p>
                            <p class="c33 c13 "><span class="c12 ">%block LatticeVectors</span></p>
                            <p class="c33 c13 "><span class="c12 ">&nbsp;1.0 &nbsp; &nbsp;0.0 &nbsp; &nbsp;0.0</span></p>
                            <p class="c33 c13 "><span class="c12 ">&nbsp;0.0 &nbsp; &nbsp;1.0 &nbsp; &nbsp;0.0</span></p>
                            <p class="c33 c13 "><span class="c12 ">&nbsp;0.0 &nbsp; &nbsp;0.0 &nbsp; &nbsp;1.0</span></p>
                            <p class="c33 c13 "><span class="c12 ">%endblock LatticeVectors</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">AtomicCoordinatesFormat &nbsp;Ang &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; # Format for coordinates</span></p>
                            <p class="c33 c13 c7 "><span class="c123 c34 c109 "></span></p>
                            <p class="c33 c13 "><span class="c41 c45 c51 ">%block AtomicCoordinatesAndAtomicSpecies</span></p>
                            <p class="c33 c13 "><span class="c51 c41 c45 ">. . .</span></p>
                            <p class="c33 c13 "><span class="c51 c41 c45 ">%endblock AtomicCoordinatesAndAtomicSpecies</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">WriteEigenvalues &nbsp; &nbsp; &nbsp;T</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">#######################################################</span></p>
                            <p class="c33 c13 "><span class="c12 "># For the following parameters, default value is ok. </span></p>
                            <p class="c33 c13 "><span class="c12 ">MeshCutoff &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 100. Ry &nbsp; &nbsp; &nbsp; # Mesh cutoff. real space mesh (Ry)</span></p>
                            <p class="c33 c13 "><span class="c12 ">PAO.BasisType &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;split &nbsp; &nbsp; &nbsp; &nbsp; # Type of PAO basis set</span></p>
                            <p class="c33 c13 "><span class="c12 ">PAO.EnergyShift &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;50 meV</span></p>
                            <p class="c33 c13 "><span class="c12 ">PAO.BasisSize &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;DZ &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# with polarization</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 "># SCF options</span></p>
                            <p class="c33 c13 "><span class="c12 ">MaxSCFIterations &nbsp; &nbsp; &nbsp; &nbsp; 60 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# Maximum number of SCF iter</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">DM.MixingWeight &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;0.1 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; # New DM amount for next SCF cycle</span></p>
                            <p class="c33 c13 "><span class="c12 ">DM.Tolerance &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 1.d-4 &nbsp; &nbsp; &nbsp; &nbsp; # Tolerance in maximum difference</span></p>
                            <p class="c33 c13 "><span class="c12 ">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# between input and output DM</span></p>
                            <p class="c33 c13 "><span class="c12 ">DM.NumberPulay &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 3</span></p>
                            <p class="c33 c13 "><span class="c12 ">DM.UseSaveDM &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; F &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; # to use continuation files</span></p>
                            <p class="c33 c13 "><span class="c12 ">MD.USeSaveXV &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; F</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">NeglNonOverlapInt &nbsp; &nbsp; &nbsp; &nbsp;false &nbsp; &nbsp; &nbsp; &nbsp; # Neglect non-overlap interactions </span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">SolutionMethod &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; diagon &nbsp; &nbsp; &nbsp; &nbsp;# OrderN or Diagon</span></p>
                            <p class="c33 c13 "><span class="c12 ">ElectronicTemperature &nbsp; &nbsp;6 K &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; # Temp. for Fermi smearing (Ry)</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">XC.Functional &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;GGA</span></p>
                            <p class="c33 c13 "><span class="c12 ">XC.Authors &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; PBE</span></p>
                            <p class="c33 c13 c7 "><span class="c34 c74 c114 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">From this we can seen that, a double zeta type basis is used along with the PBE functional for the DFT calculation. A cubic box of 25 &Aring; lengths is used to place the molecule.</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span>Run the calculation as usual and after finishing the run, plot MO energy levels, using </span><span class="c41 ">MO Energies</span><span class="c3 ">, which also indicate the level of HOMO (from the first occupied level, which is level 1):</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 262.91px;
            height: 88.62px; "><img alt=" " src="../images/image20.png " style="width: 262.91px; height: 93.23px; margin-left: -0.00px; margin-top: -4.61px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c20 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span>Then using, </span><span class="c41 ">Edit denchar3D.fdf</span><span>&nbsp;option insert the below text therein, and execute its next step, which </span><span class="c41
            ">is Run Denchar</span><span>. After finishing </span><span class="c41 ">View 2D/3D Files</span>
                <span class="c3 ">&nbsp;buttons can be clicked to view orbitals. Please see the next figures for the MOs of HOMO and LUMO (MO level, 120 and 121, respectively).</span>
            </p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <a id="t.e82eca861a899c219a5446bae8a468d47924e82f "></a>
            <a id="t.26 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c33 c13 "><span class="c12 ">#------------------------------------------------</span></p>
                            <p class="c33 c13 "><span class="c12 "># &nbsp; Note: It is a Sample Script only</span></p>
                            <p class="c33 c13 "><span class="c12 "># &nbsp; &nbsp; &nbsp; &nbsp; You have to Modify this according </span></p>
                            <p class="c33 c13 "><span class="c12 "># &nbsp; &nbsp; &nbsp; &nbsp; to your system/requirement</span></p>
                            <p class="c33 c13 "><span class="c12 "># &nbsp; &nbsp; See the Manual for more details</span></p>
                            <p class="c33 c13 "><span class="c12 ">#------------------------------------------------</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.TypeOfRun &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 3D</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.PlotCharge &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;T &nbsp; &nbsp; # If .true. SystemLabel.DM &nbsp; must be present </span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.PlotWaveFunctions &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; T &nbsp; &nbsp; # If .true. SystemLabel.WFSX must be present </span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.CoorUnits &nbsp; &nbsp; &nbsp; Ang</span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.DensityUnits &nbsp; &nbsp;Ele/Ang**3</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 "># set the mesh for wavefunction plot</span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.NumberPointsX 100</span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.NumberPointsY 100</span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.NumberPointsZ 100</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.MinX &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-12.5 Ang</span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.MaxX &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 12.5 Ang</span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.MinY &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-12.5 Ang</span></p>
                            <p class="c13 c33 "><span class="c12 ">Denchar.MaxY &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 12.5 Ang</span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.MinZ &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-12.5 Ang</span></p>
                            <p class="c33 c13 "><span class="c12 ">Denchar.MaxZ &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 12.5 Ang</span></p>
                            <p class="c33 c13 c7 "><span class="c12 "></span></p>
                            <p class="c33 c13 "><span class="c12 ">%block ChemicalSpeciesLabel</span></p>
                            <p class="c33 c13 "><span class="c12 ">&nbsp;1 &nbsp;6 &nbsp;C &nbsp; &nbsp; &nbsp;# Species index, atomic number, species label</span></p>
                            <p class="c33 c13 "><span class="c12 ">%endblock ChemicalSpeciesLabel</span></p>
                            <p class="c33 c13 c7 "><span class="c52 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 525.99px;
            height: 280.58px; "><img alt=" " src="../images/image50.png " style="width: 531.01px; height: 298.37px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 522.57px;
            height: 278.07px; "><img alt=" " src="../images/image51.png " style="width: 522.57px; height: 293.63px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 "><span class="c36 ">Figure 8.3</span><span class="c11 ">: Using </span><span class="c66 c69 c41
            ">&Gamma;</span><span>-point Siesta&rsquo;s SCF calculations yields MO coefficients. Here 120&rsquo;th wavefunction at the </span><span class="c66 c69 c41 ">&Gamma;</span><span>-point (alias </span>
                <span class="c27 c11 ">MO-120) is HOMO (upper figure) and MO-121 is LUMO. The qualities can be adjusted with Denchar parameter settings.</span>
            </p>
            <p class="c2 c13 "><span class="c80 c77 c109 c41 ">Demonstration of DJMol - 3</span></p>
            <p class="c2 c13 "><span class="c80 c58 c77 c41 ">OpenMD MD simulation of Gold-nanoparticle</span></p>
            <p class="c2 c13 c7 "><span class="c80 c58 c77 c41 "></span></p>
            <p class="c2 c13 "><span class="c3 ">In this demonstration, we would obtain the gold-gold atom pair distribution function from the OpenMD molecular dynamics trajectory. One can open OpenMD tool by:</span></p>
            <p class="c20 c13 "><span class="c42 c10 ">Execute </span><span class="c10 c95 c86 ">&#10148;</span><span class="c42 c10 ">&nbsp;OpenMD Tool</span></p>
            <p class="c2 c13 "><span>In first we have to construct a Au-nanoparticle (with </span><span class="c66 ">icosahedron</span><span class="c3 ">&nbsp;symmetry). A basic starting script for the calculation is given below (its file name is gold.omd, and note HTML styled tags in it)</span></p>
            <a id="t.6f53aaf25263948f5805fa511e897b21ce282dbe "></a>
            <a id="t.27 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c90 " colspan="1 " rowspan="1 ">
                            <p class="c33 c13 "><span class="c26 ">&lt;OpenMD&gt;</span></p>
                            <p class="c33 c13 "><span class="c26 ">&nbsp; &lt;MetaData&gt;</span></p>
                            <p class="c33 c13 c7 "><span class="c26 "></span></p>
                            <p class="c33 c13 "><span class="c26 ">molecule{</span></p>
                            <p class="c33 c13 "><span class="c26 ">name = &quot;Au&quot;;</span></p>
                            <p class="c33 c13 c7 "><span class="c26 "></span></p>
                            <p class="c33 c13 "><span class="c26 ">atom[0]{</span></p>
                            <p class="c33 c13 "><span class="c26 ">type = &quot;Au&quot;;</span></p>
                            <p class="c33 c13 "><span class="c26 ">position(0.0, 0.0, 0.0);</span></p>
                            <p class="c33 c13 "><span class="c26 ">&nbsp; &nbsp; &nbsp;}</span></p>
                            <p class="c33 c13 "><span class="c26 ">}</span></p>
                            <p class="c33 c13 c7 "><span class="c26 "></span></p>
                            <p class="c33 c13 "><span class="c26 ">component{</span></p>
                            <p class="c33 c13 "><span class="c26 ">type = &quot;Au&quot;;</span></p>
                            <p class="c33 c13 "><span class="c26 ">nMol = 1;</span></p>
                            <p class="c33 c13 "><span class="c26 ">}</span></p>
                            <p class="c33 c13 c7 "><span class="c26 "></span></p>
                            <p class="c33 c13 "><span class="c26 ">forceField = &quot;SC&quot;;</span></p>
                            <p class="c33 c13 "><span class="c26 ">forceFieldFileName = &quot;SuttonChen.QSC.frc&quot;;</span></p>
                            <p class="c33 c13 c7 "><span class="c26 "></span></p>
                            <p class="c33 c13 "><span class="c26 ">&nbsp; &lt;/MetaData&gt;</span></p>
                            <p class="c33 c13 "><span class="c34 c113 ">&lt;/OpenMD&gt;</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">Using the command (in the OpenMD tool&rsquo;s command shell window) save the file, gold.omd ,with the above data.</span></p>
            <p class="c2 c13 "><span class="c27 c41 c45 ">cmd.exe /c notepad.exe gold.omd</span></p>
            <p class="c2 c13 "><span class="c3 ">Then run the following commands one after another:</span></p>
            <p class="c2 c13 "><span class="c27 c41 c45 ">cmd.exe /c .\icosahedralBuilder.exe &nbsp;-o file.omd --shell=8 --latticeConstant=4.08 --ico gold.omd</span></p>
            <p class="c2 c13 "><span class="c27 c41 c45 ">cmd.exe /c Dump2XYZ -i file.omd</span></p>
            <p class="c2 c13 "><span class="c27 c41 c45 ">cmd.exe /c more file.xyz</span></p>
            <p class="c2 c13 "><span class="c3 ">If everything is correct you will get a gold-nanoparticle structure as shown in the below figure-a. The above procedure construct a file called, file.omd which contains all the XYZ data of the nanoparticle.</span></p>
            <p class="c2 c13 "><span class="c3 ">However in order to start an MD run, we need initial velocities. For the purpose please run the command:</span></p>
            <p class="c2 c13 "><span class="c27 c41 c45 ">cmd.exe /c .\thermalizer -t 5 -o file-5K.omd file.omd</span></p>
            <p class="c2 c13 "><span class="c3 ">It will create a new file, file-5K.omd with new initial velovities and 5 Kelvin as its target temperature. Its main parameters are:</span></p>
            <a id="t.245ed061811ea654de86b291ec7f8e5acd6541bd "></a>
            <a id="t.28 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c90 " colspan="1 " rowspan="1 ">
                            <p class="c33 c13 c7 "><span class="c3 "></span></p>
                            <p class="c33 c13 "><span class="c26 ">forceField = &quot;SC&quot;;</span></p>
                            <p class="c33 c13 "><span class="c26 ">forceFieldFileName = &quot;SuttonChen.QSC.frc&quot;;</span></p>
                            <p class="c33 c13 c7 "><span class="c26 "></span></p>
                            <p class="c33 c13 "><span class="c26 ">ensemble = &quot;LHull&quot;;</span></p>
                            <p class="c33 c13 "><span class="c26 ">targetTemp = 5;</span></p>
                            <p class="c33 c13 "><span class="c26 ">targetPressure = 1;</span></p>
                            <p class="c33 c13 "><span class="c26 ">viscosity = 0.1;</span></p>
                            <p class="c33 c13 c7 "><span class="c26 "></span></p>
                            <p class="c33 c13 "><span class="c26 ">dt = 2.0;</span></p>
                            <p class="c33 c13 "><span class="c26 ">runTime = 2E5;</span></p>
                            <p class="c33 c13 "><span class="c26 ">sampleTime = 500.0;</span></p>
                            <p class="c33 c13 "><span class="c26 ">statusTime = 4;</span></p>
                            <p class="c33 c13 "><span class="c26 ">seed = 985456376;</span></p>
                            <p class="c33 c13 "><span class="c26 ">usePeriodicBoundaryConditions = &quot;false&quot;;</span></p>
                            <p class="c33 c13 "><span class="c26 ">tauThermostat = 1E3;</span></p>
                            <p class="c33 c13 "><span class="c34 c113 ">tauBarostat = 5E3;</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">Then run the actual OpenMD molecular dynamics run as,</span></p>
            <p class="c2 c13 "><span class="c27 c41 c45 ">cmd.exe /c openmd file-5K.omd</span></p>
            <p class="c2 c13 "><span class="c3 ">This may take a while (2 hour), and its trajectory information will stored in the dump file. From this MD animation can be retrived as:</span></p>
            <p class="c2 c13 "><span class="c27 c41 c45 ">cmd.exe /c Dump2XYZ -b &nbsp;-i file-5K.dump</span></p>
            <p class="c2 c13 "><span class="c6 ">Moreover, from the below figure one can see that the 5K thermalization is achieved in the MD run (note its convergence to 5K)</span></p>
            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 495.27px;
            height: 268.80px; "><img alt=" " src="../images/image52.png " style="width: 495.27px; height: 268.80px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span>After that, a script utility to retrieve, </span><span class="c66 c86 c128 ">g</span><span class="c128 c62 c66 c86 ">Au-Au(</span><span class="c128 c66 c86 ">(r), </span><span class="c3 ">as:</span></p>
            <p class="c2 c13 "><span class="c27 c41 c45 ">cmd.exe /c StaticProps.exe -i file-5K.dump --gofr --sele1=&quot;select Au*&quot; --sele2=&quot;select Au*&quot;</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 329.73px;
            height: 211.51px; "><img alt=" " src="../images/image53.png " style="width: 329.73px; height: 211.51px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c30 c13 "><span class="c36 ">Figure 8.4: </span><span class="c11 ">Pair correlation function (</span><span class="c1 ">g(r)</span><span class="c11 ">) obtained from the MD trajectory data which use only test-level parameters. Note the similarity with </span>
                <span class="c1 ">g(r)</span><span class="c11 ">&nbsp;obtained by a hard-sphere model (inset figure courtesy, </span><span class="c1 ">J. Nanopart. Res, 13, 4277 (2011).</span><span class="c27 c11 ">)</span></p>
            <p class="c30 c13 c7 "><span class="c27 c11 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 282.01px;
            height: 149.45px; "><img alt=" " src="../images/image54.png " style="width: 282.01px; height: 158.65px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 283.28px;
            height: 150.76px; "><img alt=" " src="../images/image55.png " style="width: 283.28px; height: 159.36px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 "><span class="c41 ">Figure 8.5</span><span class="c3 ">: The initial (top) and final (bottom) snapshot of the MD run. Note that the system&rsquo;s initial temperature was 0K and it was equilibrated to 5K during the MD run.</span></p>
            <p class="c2 c13 "><span class="c80 c77 c109 c41 ">Demonstration of DJMol - 4</span></p>
            <p class="c2 c13 "><span class="c80 c58 c77 c41 ">ASE simulation of an adatom diffusion</span></p>
            <p class="c2 c13 "><span>Here we use ASE to obtain the energy profile of a physical process - </span><span class="c66 ">diffusion of an adatom</span><span class="c3 ">&nbsp;or adsorbed atom (Au) on the top of a metallic surface of Al atoms. Nudged elastic band (NEB) method is used for the calculation, in which an initial structure (where an adatom i.e. gold atom is placed in the left side) and a final structure (adatom is placed in the right side) of the process is used as its input data. For more details please visit ASE tutorial web page. Note that we used EMT calculators (it is a really fast albeit highly approximated potential function) for this calculation however one can use Siesta within ASE to obtain more realistic diffusion barrier.</span></p>
            <p class="c2 c13 "><span>Refer Chapter-5 for creating a </span><span class="c41 ">Python Project</span><span>. Then add an </span><span class="c41
            ">Empty module</span><span>&nbsp;(InitialFinalStructure.py fie) into the </span><span class="c41 ">src</span>
                <span class="c3 ">&nbsp;directory. Then add the following Python source into it:</span>
            </p>
            <a id="t.3f4d612311a01c0fccc926a4b5fe7f3bd41f8c2f "></a>
            <a id="t.29 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c90 " colspan="1 " rowspan="1 ">
                            <p class="c39 c13 c7 "><span class="c50 "></span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.build import fcc100, add_adsorbate</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.constraints import FixAtoms</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.calculators.emt import EMT</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.optimize import QuasiNewton</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># 3x3-Al(001) surface with 3 layers and an</span></p>
                            <p class="c39 c13 "><span class="c26 "># Au atom adsorbed in a hollow site:</span></p>
                            <p class="c39 c13 "><span class="c26 ">slab = fcc100(&#39;Al&#39;, size=(3, 3, 3))</span></p>
                            <p class="c39 c13 "><span class="c26 ">add_adsorbate(slab, &#39;Au&#39;, 1.7, &#39;hollow&#39;)</span></p>
                            <p class="c39 c13 "><span class="c26 ">slab.center(axis=2, vacuum=4.0)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Fix second and third layers:</span></p>
                            <p class="c39 c13 "><span class="c26 ">mask = [atom.tag &gt; 1 for atom in slab]</span></p>
                            <p class="c39 c13 "><span class="c26 ">slab.set_constraint(FixAtoms(mask=mask))</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Use EMT potential:</span></p>
                            <p class="c39 c13 "><span class="c26 ">slab.calc = EMT()</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Initial state:</span></p>
                            <p class="c39 c13 "><span class="c26 ">qn = QuasiNewton(slab, trajectory=&#39;initial.traj&#39;)</span></p>
                            <p class="c39 c13 "><span class="c26 ">qn.run(fmax=0.05)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Final state:</span></p>
                            <p class="c39 c13 "><span class="c26 ">slab[-1].x += slab.get_cell()[0, 0] / 3</span></p>
                            <p class="c39 c13 "><span class="c26 ">qn = QuasiNewton(slab, trajectory=&#39;final.traj&#39;)</span></p>
                            <p class="c39 c13 "><span class="c26 ">qn.run(fmax=0.05)</span></p>
                            <p class="c33 c13 c7 "><span class="c50 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span>Then Run the file (Run </span><span class="c11 ">&#9658;</span><span class="c3 ">&nbsp;Run File). This will create, initial.traj and final.traj files which contains the initial and final structure respectively.</span></p>
            <p class="c2 c13 "><span class="c3 ">After this, the NEB calculation can be started (don&rsquo;t forget to create a new Python file for this) and its Python source is given below:</span></p>
            <a id="t.79106bbe5e685ec917f774ef1067a44352299680 "></a>
            <a id="t.30 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c25 c13 "><span class="c26 "># To change this license header, choose License Headers in Project Properties.</span></p>
                            <p class="c25 c13 "><span class="c26 "># To change this template file, choose Tools | Templates</span></p>
                            <p class="c25 c13 "><span class="c26 "># and open the template in the editor.</span></p>
                            <p class="c25 c13 "><span class="c26 ">from ase.io import read</span></p>
                            <p class="c25 c13 "><span class="c26 ">from ase.constraints import FixAtoms</span></p>
                            <p class="c25 c13 "><span class="c26 ">from ase.calculators.emt import EMT</span></p>
                            <p class="c25 c13 "><span class="c26 ">from ase.neb import NEB</span></p>
                            <p class="c25 c13 "><span class="c26 ">from ase.optimize import BFGS</span></p>
                            <p class="c25 c13 c7 "><span class="c26 "></span></p>
                            <p class="c25 c13 "><span class="c26 ">initial = read(&#39;initial.traj&#39;)</span></p>
                            <p class="c25 c13 "><span class="c26 ">final = read(&#39;final.traj&#39;)</span></p>
                            <p class="c25 c13 "><span class="c26 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p>
                            <p class="c25 c13 "><span class="c26 ">constraint = FixAtoms(mask=[atom.tag &gt; 1 for atom in initial])</span></p>
                            <p class="c25 c13 c7 "><span class="c26 "></span></p>
                            <p class="c25 c13 "><span class="c26 ">images = [initial]</span></p>
                            <p class="c25 c13 "><span class="c26 ">for i in range(9):</span></p>
                            <p class="c25 c13 "><span class="c26 ">&nbsp; &nbsp; image = initial.copy()</span></p>
                            <p class="c25 c13 "><span class="c26 ">&nbsp; &nbsp; image.calc = EMT()</span></p>
                            <p class="c25 c13 "><span class="c26 ">&nbsp; &nbsp; image.set_constraint(constraint)</span></p>
                            <p class="c25 c13 "><span class="c26 ">&nbsp; &nbsp; images.append(image)</span></p>
                            <p class="c25 c13 c7 "><span class="c26 "></span></p>
                            <p class="c25 c13 "><span class="c26 ">images.append(final)</span></p>
                            <p class="c25 c13 c7 "><span class="c26 "></span></p>
                            <p class="c25 c13 "><span class="c26 ">neb = NEB(images)</span></p>
                            <p class="c25 c13 "><span class="c26 ">neb.interpolate()</span></p>
                            <p class="c25 c13 "><span class="c26 ">qn = BFGS(neb, trajectory=&#39;neb.traj&#39;)</span></p>
                            <p class="c25 c13 "><span class="c26 ">qn.run(fmax=0.05)</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">The energy profile can be saved in a PNG image by using the following script:</span></p>
            <a id="t.ffbd57a14d7a18fd50794bb6da706d4438e587fb "></a>
            <a id="t.31 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c39 c13 "><span class="c26 ">import matplotlib.pyplot as plt</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.neb import NEBTools</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.io import read</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 ">images = read(&#39;neb.traj@-11:&#39;)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 ">nebtools = NEBTools(images)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Get the calculated barrier and the energy change of the reaction.</span></p>
                            <p class="c39 c13 "><span class="c26 ">Ef, dE = nebtools.get_barrier()</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Get the barrier without any interpolation between highest images.</span></p>
                            <p class="c39 c13 "><span class="c26 ">Ef, dE = nebtools.get_barrier(fit=False)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Get the actual maximum force at this point in the simulation.</span></p>
                            <p class="c39 c13 "><span class="c26 ">max_force = nebtools.get_fmax()</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Create a figure like that coming from ASE-GUI.</span></p>
                            <p class="c39 c13 "><span class="c26 ">fig = nebtools.plot_band()</span></p>
                            <p class="c39 c13 "><span class="c26 ">fig.savefig(&#39;diffusion-barrier.png&#39;)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># Create a figure with custom parameters.</span></p>
                            <p class="c39 c13 "><span class="c26 ">fig = plt.figure(figsize=(5.5, 4.0))</span></p>
                            <p class="c39 c13 "><span class="c26 ">ax = fig.add_axes((0.15, 0.15, 0.8, 0.75))</span></p>
                            <p class="c39 c13 "><span class="c26 ">nebtools.plot_band(ax)</span></p>
                            <p class="c39 c13 "><span class="c26 ">fig.savefig(&#39;diffusion-barrier.png&#39;)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c33 c13 c7 "><span class="c26 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">Finally, the neb.traj file, which contains the entire geometry of the process in binary format, is then used to obtain XYZ format by the following script:</span></p>
            <a id="t.3ea317478f8e68fd3d1e2ecdb74eb92ac54d8f93 "></a>
            <a id="t.32 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c39 c13 "><span class="c26 ">import matplotlib.pyplot as plt</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.calculators.emt import EMT</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.neb import NEB</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.optimize import BFGS</span></p>
                            <p class="c39 c13 "><span class="c26 ">from ase.io import read, write</span></p>
                            <p class="c39 c13 "><span class="c26 ">import os</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># read the last structures (of 5 images used in NEB)</span></p>
                            <p class="c39 c13 "><span class="c26 ">images = read(&#39;neb.traj@-11:&#39;)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 ">for i in range(0, &nbsp;len(images)):</span></p>
                            <p class="c39 c13 "><span class="c26 ">&nbsp; &nbsp; atoms = images[i]</span></p>
                            <p class="c39 c13 "><span class="c26 ">&nbsp; &nbsp; #print(atoms.get_positions())</span></p>
                            <p class="c39 c13 "><span class="c26 ">&nbsp; &nbsp; write(&#39;delete.xyz&#39;, atoms)</span></p>
                            <p class="c39 c13 "><span class="c26 ">&nbsp; &nbsp; inFile = open(os.path.join( &#39;delete.xyz&#39;), &#39;r&#39;)</span></p>
                            <p class="c39 c13 "><span class="c26 ">&nbsp; &nbsp; fileStr = inFile.read()</span></p>
                            <p class="c13 c39 "><span class="c26 ">&nbsp; &nbsp; outFile = open(&quot;neb_movie.xyz&quot;, &#39;a&#39;)</span></p>
                            <p class="c39 c13 "><span class="c26 ">&nbsp; &nbsp; outFile.write(fileStr)</span></p>
                            <p class="c39 c13 c7 "><span class="c26 "></span></p>
                            <p class="c39 c13 "><span class="c26 "># @type outFile </span></p>
                            <p class="c39 c13 "><span class="c26 ">outFile.close()</span></p>
                            <p class="c33 c13 "><span class="c26 ">inFile.close()</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span>Additionally, using DJMol&#39;s xyz File loader (File</span><span class="c11 ">&#9658;</span><span class="c3 ">Open Structure) the geometries of the process as it is saved in &quot;neb_movie.xyz&quot; can be displayed or animated (See the below figure).</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 496.07px;
            height: 215.87px; "><img alt=" " src="../images/image56.png " style="width: 496.07px; height: 215.87px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c30 c13 "><span class="c36 ">Figure 8.6:</span><span class="c27 c11 ">&nbsp;Diffusion of gold atom over the aluminium surface (FCC) from one hollow site to another, in different perspectives; Nine NEB images were used for this NEB profile.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 494.60px;
            height: 153.53px; "><img alt=" " src="../images/image57.png " style="width: 494.60px; height: 153.53px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c30 c13 "><span class="c36 ">Figure 8.7:</span><span class="c27 c11 ">&nbsp;Starting (I) and ending (F) geometries of the Diffusion of gold atom; The diffusion barrier (around 0.35 eV) is shown in the inset.</span></p>

            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c80 c58 c77 c41 "></span></p>
            <p class="c2 c13 c7 "><span class="c80 c58 c77 c41 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
        </div>
        <div id="appendices">
            <p class="c2 c13 c7 "><span class="c3 "></span></p>Appendic
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <h4 class="c92 c13 "><span class="c63 c58 c77 c86 "></span></h4>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <h4 class="c13 c132 "><span class="c58 c77 c41 c98 ">Appendices</span></h4>
            <h4 class="c92 c13 "><span class="c63 c58 c77 c86 "></span></h4>
            <h4 class="c13 c92 "><span class="c63 c58 c77 c86 "></span></h4>
            <h4 class="c92 c13 "><span class="c63 c58 c77 c86 "></span></h4>
            <h4 class="c92 c13 "><span class="c63 c58 c77 c86 "></span></h4>
            <h4 class="c92 c13 "><span class="c63 c58 c77 c86 "></span></h4>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c65 c13 c7 subtitle " id="h.35nkun2 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c13 c7 c60 subtitle "><span class="c3 "></span></p>
            <div id="appendix-a"></div>
            <p class="c65 c13 subtitle "><span class="c6 ">Appendix-A</span></p>
            <h1 class="c137 c13 " id="h.1ksv4uv "><span class="c36 c58 c68 ">Python Scripting for DJMol Applications</span></h1>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Since DJMol uses Python for its scripting (and in ASE), a basic knowledge of Python (version 3) is highly appreciated. This appendix is based on the book, </span><span class="c46 c41
            ">Think Python </span><span class="c6 ">2nd Edition by Allen B. Downey (Creative Commons Attribution-NonCommercial 3.0 Unported License). Some familiarity with programming is assumed for this appendix.</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c41 c43 ">How to use Python within the DJMol?</span></p>
            <p class="c2 c13 "><span class="c6 ">To use DJMol, Python is a mandate tool and it should be installed. For this, you have to install the Python (windows version 3 or later) immediately after the installation of the DJMol program into your system. See Appendix - 3 for its details (including the details of running a python in the DJMol) </span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c43 c41 ">Basics of python scripting</span></p>
            <p class="c2 c13 c7 "><span class="c43 c41 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Python is designed as an interpreter language and it is widely used in scripting purposes. It supports both procedural and object-oriented paradigm. </span></p>
            <p class="c2 c13 c7 "><span class="c43 c41 "></span></p>
            <p class="c2 c13 "><span class="c4 ">Variable assignment and Operators</span></p>
            <p class="c33 c7 c57 "><span class="c6 "></span></p>
            <p class="c33 c57 "><span class="c6 ">In Python variable assignment is performed by, </span><span class="c4 ">= </span><span class="c6 ">operator. It also provides operators, which are special symbols that represent operations like addition or multiplication. The operators </span>
                <span class="c4 ">+</span><span class="c6 ">, </span><span class="c4 ">-</span><span class="c6 ">, </span><span class="c4 ">/, </span><span class="c6 ">and</span><span class="c4 ">&nbsp;* </span><span class="c6 ">perform addition, subtraction, division, and multiplication, respectively. And the operator </span>
                <span class="c4 ">** </span><span class="c6 ">performs exponentiation; that is, it raises a number to a power. An illustration (hopefully self-explanatory!) of these commands are following (here, </span><span class="c152 c42 c155 ">&gt;&gt;&gt;</span>
                <span class="c6 ">&nbsp;string represents a command prompt; for the comment one can use #).</span>
            </p>
            <p class="c33 c7 "><span class="c3 "></span></p>
            <a id="t.a94e7da40ceec0a6c752ff07c8441c6928927544 "></a>
            <a id="t.33 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x=5</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x</span></p>
                            <p class="c2 "><span class="c5 ">5</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x+2 # a comment </span></p>
                            <p class="c2 "><span class="c5 ">7</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x-2</span></p>
                            <p class="c2 "><span class="c5 ">3</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x*2</span></p>
                            <p class="c2 "><span class="c5 ">10</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x**2</span></p>
                            <p class="c2 "><span class="c5 ">25</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x%2</span></p>
                            <p class="c2 "><span class="c5 ">1</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x/2</span></p>
                            <p class="c2 "><span class="c5 ">2.5</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c33 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Note that in numerical computation one need to use integers and decimals. Though it doesn&#39;t require explicit declaration of type of the variables, there may be times when you want to specify the type of a variable. This can be done with casting operation. </span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Casting in python is therefore done using constructor functions:</span></p>
            <ul class="c112 lst-kix_list_2-0 start ">
                <li class="c75 c9 c59 li-bullet-0 "><span class="c6 ">int() - constructs an integer number from an integer literal, a float literal (by rounding down to the previous whole number), or a string literal (providing the string represents a whole number)</span></li>
                <li class="c75 c9 c59 li-bullet-0 "><span class="c6 ">float() - constructs a float number from an integer literal, a float literal or a string literal (providing the string represents a float or an integer)</span></li>
                <li class="c75 c9 c59 li-bullet-0 "><span class="c6 ">str() - constructs a string from a wide variety of data types, including strings, integer literals and float literals</span></li>
            </ul>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <a id="t.523d4b54b1f7277fc97bf696c213dd67f85c6e3b "></a>
            <a id="t.34 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; a=1</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; type(a)</span></p>
                            <p class="c2 "><span class="c5 ">&lt;class &#39;int&#39;&gt;</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; b=3.1415</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; type(b)</span></p>
                            <p class="c2 "><span class="c5 ">&lt;class &#39;float&#39;&gt;</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; a+b</span></p>
                            <p class="c2 "><span class="c5 ">4.141500000000001</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; b**b</span></p>
                            <p class="c2 "><span class="c5 ">36.45491472872008</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; y = int(2.8)</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; y</span></p>
                            <p class="c2 "><span class="c5 ">2</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x = float(1)</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; x</span></p>
                            <p class="c2 "><span class="c5 ">1.0</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; z = float(&quot;3&quot;)</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; z</span></p>
                            <p class="c2 c13 "><span class="c5 ">3.0</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c27 c77 c41 "></span></p>
            <p class="c2 c13 "><span class="c43 c41 ">String, List, Methods etc.</span></p>
            <p class="c2 c13 c7 "><span class="c4 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Apart from </span><span class="c46 c41 ">int</span><span class="c6 ">, </span><span class="c46 c41 ">string </span><span class="c6 ">type is also used in python. Its initiation is simple (you can use either &lsquo; or &ldquo; to make string):</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <a id="t.e993ff7846118067a9fd4c9f08ed17e22e3ca5df "></a>
            <a id="t.35 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_string = &#39;thisStringI&#39;</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_string</span></p>
                            <p class="c2 "><span class="c5 ">&#39;thisString&#39;</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_string = &quot;thisString&quot;</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_string</span></p>
                            <p class="c2 "><span class="c5 ">&#39;thisString&#39;</span></p>
                            <p class="c2 c13 c7 "><span class="c6 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c27 c77 c41 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Like a string, a list is a sequence of values (perhaps it may be the Pythons most important built-in data type; unlike C or Fortran python does not have an </span><span class="c46 c41
            ">Array </span><span class="c6 ">data type). In a string, the values are characters; </span>
                <span class="c0 ">in a </span><span class="c46 c41 ">list, </span><span class="c0 ">they can be </span><span class="c46 c41 ">any type</span><span class="c6 ">. The values in a list are called </span><span class="c46 c41 ">elements </span>
                <span class="c6 ">or sometimes </span>
                <span class="c46 c41 ">items</span><span class="c6 ">. There are several ways to create a new list; the simplest is to enclose the elements in square brackets:</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <a id="t.0f6213d5cc40625976b3651dadce9ce43a9b40bf "></a>
            <a id="t.36 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; a = &#39;is&#39;</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; b = &#39;nice&#39;</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list = [&#39;my&#39;, &#39;list&#39;, a, b]</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list2 = [[4,5,6,7], [3,4,5,6]]</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list[0]</span></p>
                            <p class="c2 "><span class="c5 ">&#39;my&#39;</span></p>
                            <p class="c2 c13 c7 "><span class="c3 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Note that a list can be used as an Array (and its index starts from 0 and not from 1, like a C array).</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c43 c41 ">List Manipulation Rules</span></p>
            <a id="t.7fbd2ddcf67855c81dc02f860e582f39f4397bb4 "></a>
            <a id="t.37 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c24 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list[0]</span></p>
                            <p class="c2 "><span class="c5 ">&#39;my&#39;</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list[1:3]</span></p>
                            <p class="c2 "><span class="c5 ">[&#39;list&#39;, &#39;is&#39;]</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list[1:]</span></p>
                            <p class="c2 "><span class="c5 ">[&#39;list&#39;, &#39;is&#39;, &#39;nice&#39;]</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list[:3]</span></p>
                            <p class="c2 "><span class="c5 ">[&#39;my&#39;, &#39;list&#39;, &#39;is&#39;]</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list[:]</span></p>
                            <p class="c2 "><span class="c5 ">[&#39;my&#39;, &#39;list&#39;, &#39;is&#39;, &#39;nice&#39;]</span></p>
                            <p class="c2 c13 c7 "><span class="c43 c41 "></span></p>
                        </td>
                        <td class="c14 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c3 ">Select the first item</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Select items at index 1 and 2</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Select items after index 0</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Select items before index 3</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c13 "><span class="c3 ">Copy my_list</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c43 c41 "></span></p>
            <p class="c2 c13 "><span class="c43 c41 ">List Operations</span></p>
            <a id="t.f65aa3d94facedce415cc85cc79cc4e0319e5d5e "></a>
            <a id="t.38 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list + my_list</span></p>
                            <p class="c2 "><span class="c5 ">[&#39;my&#39;, &#39;list&#39;, &#39;is&#39;, &#39;nice&#39;, &#39;my&#39;, &#39;list&#39;, &#39;is&#39;, &#39;nice&#39;]</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list * 2</span></p>
                            <p class="c2 "><span class="c5 ">[&#39;my&#39;, &#39;list&#39;, &#39;is&#39;, &#39;nice&#39;, &#39;my&#39;, &#39;list&#39;, &#39;is&#39;, &#39;nice&#39;]</span></p>
                            <p class="c2 c13 c7 "><span class="c3 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c43 c41 ">Methods for the Lists </span></p>
            <p class="c2 c13 "><span class="c6 ">The Methods are equivalent to functions/subroutines; Since the </span><span class="c46 c41 ">List </span><span class="c6 ">is an object a method for that list is called by a </span><span class="c4
            ">DOT (.) </span><span class="c6 ">operator immediately after the Object name, as it is illustrated below:</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <a id="t.421c212bb4a48e586f17d8983928f673815e16c0 "></a>
            <a id="t.39 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c24 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list.index(a)</span></p>
                            <p class="c2 "><span class="c5 ">2</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list.count(a)</span></p>
                            <p class="c2 "><span class="c5 ">1</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list.append(&#39;NewOne&#39;)</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list</span></p>
                            <p class="c2 "><span class="c5 ">[&#39;my&#39;, &#39;list&#39;, &#39;is&#39;, &#39;nice&#39;, &#39;NewOne&#39;]</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list.remove(&#39;is&#39;)</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; my_list</span></p>
                            <p class="c2 "><span class="c5 ">[&#39;my&#39;, &#39;list&#39;, &#39;nice&#39;, &#39;NewOne&#39;]</span></p>
                            <p class="c2 c13 c7 "><span class="c6 "></span></p>
                        </td>
                        <td class="c14 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c3 ">Get the index of an item </span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Count an item </span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Append an item at a time</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">It shows the appended List</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c13 "><span class="c3 ">Remove an Item from the list</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c43 c41 ">Libraries</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">In Python a good collection of libraries are available say, for doing math, or to do string/array manipulation, Numerical computations, graphics etc.</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">The math library of python can be imported like:</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <a id="t.c0fde1c8497e36678a99b11afc19cd5e7afe1a68 "></a>
            <a id="t.40 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c87 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; import math</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; math.sqrt(9.99)</span></p>
                            <p class="c2 "><span class="c5 ">3.1606961258558215</span></p>
                            <p class="c2 c13 c7 "><span class="c6 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">For ASE scripting, most of the times you need to use NumPy and Scipy libraries (and it is not shipped with standard Python, so one needs to install it, see the next appendix). If you installed NumPy you can call it like:</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <a id="t.c58956edc7a800d4c97cd6d4c651b230e13fc204 "></a>
            <a id="t.41 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c24 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt;import numpy</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">Or </span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt;import numpy as np</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; Array=np.array([[1, 2], [3, 4]])</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; Array</span></p>
                            <p class="c2 "><span class="c5 ">array([[1, 2],</span></p>
                            <p class="c2 "><span class="c5 ">&nbsp; &nbsp; &nbsp; &nbsp;[3, 4]])</span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; np.transpose(Array)</span></p>
                            <p class="c2 "><span class="c5 ">array([[1, 3],</span></p>
                            <p class="c2 "><span class="c5 ">&nbsp; &nbsp; &nbsp; &nbsp;[2, 4]])</span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&gt;&gt;&gt; Array.dot(Array)</span></p>
                            <p class="c2 "><span class="c5 ">array([[ 7, 10],</span></p>
                            <p class="c2 c13 "><span class="c5 ">&nbsp; &nbsp; &nbsp; &nbsp;[15, 22]])</span></p>
                        </td>
                        <td class="c14 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c3 ">Importing NumPy Lib.</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Constructing an Array</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Print that Array</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Transpose of the Matrix</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c13 "><span class="c3 ">Matrix Multiplication</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c43 c41 ">Basics of ASE scripting</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Hopefully one can be now understood ASE scripting. A sample ASE script is given below with some explanations. The aim is to calculate N2 molecular energy with EMT calculator (Effective Medium Potential, a crude empirical model generally used for testing purposes or for very large scale MD). Note that the following code has six lines (didn&rsquo;t use line-breaker)</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <a id="t.69a51246b182418adef3098ff3bb4a8a101a369c "></a>
            <a id="t.42 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c179 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c5 ">from ase import Atoms </span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">from ase.calculators.emt import EMT </span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">d = 1.1 # Bond Length in Angstrom</span></p>
                            <p class="c2 "><span class="c5 ">molecule = Atoms(&#39;2N&#39;, [(0., 0., 0.), (0., 0., d)]) </span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">molecule.set_calculator(EMT()) </span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">e_molecule = molecule.get_potential_energy() </span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 c7 "><span class="c5 "></span></p>
                            <p class="c2 "><span class="c5 ">&nbsp;</span></p>
                            <p class="c2 "><span class="c5 ">print (&#39;Nitrogen molecule energy: %5.2f eV&#39; % e_molecule) </span></p>
                            <p class="c2 c13 c7 "><span class="c6 "></span></p>
                        </td>
                        <td class="c116 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c3 ">Import Atoms object</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Import EMT </span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">Define a </span><span class="c27 c77 c41 ">molecule </span><span class="c3 ">with Atoms Object</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c3 ">That molecule is </span><span class="c27 c77 c41 ">attached </span><span class="c3 ">with EMT</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 "><span class="c27 c77 c41 ">Calculate PE</span></p>
                            <p class="c2 "><span class="c3 ">(this Really invoke the EMT calculations)</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                            <p class="c2 c13 "><span class="c3 ">Printing the PE</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Note that the important step is to create the </span><span class="c4 ">Atoms </span><span class="c6 ">object - which is a collection of atoms. Everything else is based on this </span><span class="c4 ">Atoms object (</span>
                <span class="c6 ">for example DFT </span>
                <span class="c4 ">calculators </span><span class="c6 ">can be attached to this object to get DFT total energy</span><span class="c4 ">).</span></p>
            <p class="c2 c13 c7 "><span class="c4 "></span></p>
            <p class="c2 c13 "><span class="c6 ">A little more advanced example is following (an infinite gold wire with BL= 2.9 A).</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <a id="t.02e5f483f2751c2fdfbb3b2f963846fdbe642da2 "></a>
            <a id="t.43 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c163 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c91 c96 c41 c45 ">from</span><span class="c40 c34 ">&nbsp;</span><span class="c91 c41 c45 c181 ">ase</span><span class="c40 c34 ">&nbsp;</span><span class="c91 c41 c45 c96
            ">import</span><span class="c40 c34 ">&nbsp;</span>
                                <span class="c40 c144 c41 c45 ">Atoms</span>
                            </p>
                            <p class="c2 "><span class="c40 c34 ">d </span><span class="c34 c49 ">=</span><span class="c40 c34 ">&nbsp;</span><span class="c35 c34 ">2.9</span></p>
                            <p class="c2 "><span class="c40 c34 ">L </span><span class="c49 c34 ">=</span><span class="c40 c34 ">&nbsp;</span><span class="c35 c34 ">10.0</span></p>
                            <p class="c2 "><span class="c40 c34 ">wire </span><span class="c49 c34 ">=</span><span class="c40 c34 ">&nbsp;</span><span class="c40 c41 c45 c144 ">Atoms</span><span class="c40 c34 ">(</span><span class="c91 c34 c136 ">&#39;Au&#39;</span>
                                <span class="c40 c34 ">,</span>
                            </p>
                            <p class="c2 "><span class="c40 c34 ">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;positions</span><span class="c49 c34 ">=</span><span class="c40 c34 ">[[</span><span class="c35 c34 ">0</span><span class="c40 c34
            ">, L </span><span class="c49 c34 ">/</span>
                                <span class="c40 c34 ">&nbsp;</span><span class="c35 c34 ">2</span><span class="c40 c34 ">, L </span><span class="c49 c34 ">/</span><span class="c40 c34 ">&nbsp;</span><span class="c34 c35 ">2</span><span class="c40 c34 ">]],</span></p>
                            <p class="c2 "><span class="c40 c34 ">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;cell</span><span class="c49 c34 ">=</span><span class="c40 c34 ">[d, L, L],</span></p>
                            <p class="c54 "><span class="c40 c34 ">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;pbc</span><span class="c49 c34 ">=</span><span class="c40 c34 ">[</span><span class="c35 c34 ">1</span><span class="c40 c34
            ">, </span><span class="c35 c34 ">0</span>
                                <span class="c40 c34 ">, </span><span class="c35 c34 ">0</span><span class="c40 c34 ">])</span></p>
                            <p class="c2 c13 c7 "><span class="c6 "></span></p>
                        </td>
                        <td class="c130 " colspan="1 " rowspan="1 ">
                            <p class="c2 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            width: 269.00px; height: 186.00px; "><img alt=" " src="../images/image47.png " style="width: 269.00px; height: 186.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 "><span class="c6 ">Courtesy: ASE Camd</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">A list of </span><span class="c46 c41 ">generally used </span><span class="c6 ">GET/SET methods for the Atoms Object is:</span></p>
            <a id="t.f08f0ac0e2a9debd2852438188263eec7ec20474 "></a>
            <a id="t.44 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c24 " colspan="1 " rowspan="1 ">
                            <p class="c2 c13 "><span class="c3 ">GET Methods</span></p>
                        </td>
                        <td class="c14 " colspan="1 " rowspan="1 ">
                            <p class="c2 c13 "><span class="c3 ">SET Methods</span></p>
                        </td>
                    </tr>
                    <tr class="c15 ">
                        <td class="c24 " colspan="1 " rowspan="1 ">
                            <p class="c89 "><span class="c3 ">get_atomic_numbers()</span></p>
                            <p class="c89 "><span class="c3 ">get_initial_charges()</span></p>
                            <p class="c89 "><span class="c3 ">get_charges()</span></p>
                            <p class="c89 "><span class="c3 ">get_chemical_symbols()</span></p>
                            <p class="c89 "><span class="c3 ">get_initial_magnetic_moments()</span></p>
                            <p class="c89 "><span class="c3 ">get_magnetic_moments()</span></p>
                            <p class="c89 "><span class="c3 ">get_masses()</span></p>
                            <p class="c89 "><span class="c3 ">get_momenta()</span></p>
                            <p class="c89 "><span class="c3 ">get_forces()</span></p>
                            <p class="c89 "><span class="c3 ">get_positions()</span></p>
                            <p class="c89 "><span class="c3 ">get_potential_energies()</span></p>
                            <p class="c89 "><span class="c3 ">get_scaled_positions()</span></p>
                            <p class="c89 "><span class="c3 ">get_stresses()</span></p>
                            <p class="c89 "><span class="c3 ">get_tags()</span></p>
                            <p class="c89 "><span class="c3 ">get_velocities()</span></p>
                            <p class="c2 c13 c7 "><span class="c6 "></span></p>
                        </td>
                        <td class="c14 " colspan="1 " rowspan="1 ">
                            <p class="c2 "><span class="c3 ">set_atomic_numbers()</span></p>
                            <p class="c2 "><span class="c3 ">set_initial_charges()</span></p>
                            <p class="c2 "><span class="c3 ">set_chemical_symbols()</span></p>
                            <p class="c2 "><span class="c3 ">set_initial_magnetic_moments()</span></p>
                            <p class="c2 "><span class="c3 ">set_masses()</span></p>
                            <p class="c2 "><span class="c3 ">set_momenta()</span></p>
                            <p class="c2 "><span class="c3 ">set_positions()</span></p>
                            <p class="c2 "><span class="c3 ">set_scaled_positions()</span></p>
                            <p class="c2 "><span class="c3 ">set_tags()</span></p>
                            <p class="c2 "><span class="c3 ">set_velocities()</span></p>
                            <p class="c2 c7 "><span class="c3 "></span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 " id="h.44sinio "><span class="c6 ">In essence, the </span><span class="c4 ">Set methods</span><span class="c6 ">&nbsp;are used to supply (the necessary) information for the </span><span class="c4
            ">Atoms</span><span class="c6 ">&nbsp;object, whereas </span>
                <span class="c4 ">Get methods</span>
                <span class="c6 ">&nbsp;are applied to get useful information (usually after setting a calculator)</span>
            </p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <div id="appendix-b"></div>
            <p class="c13 c65 subtitle " id="h.2jxsxqh "><span class="c6 ">Appendix-B</span></p>
            <h1 class="c13 c137 "><span class="c68 c36 c58 ">Installation of Python and NumPy</span></h1>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">As it is said before, one need to install Python (3.x, 64 bit) scripting language and NumPy on the system </span><span class="c46 c41 ">before </span><span class="c6 ">the DJMol installation.</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c21 c13 "><span class="c6 ">Using this tutorial, one is expected to get sufficient information on how to install a </span><span class="c4 ">64 bit</span><span class="c6 ">&nbsp;version of </span><span class="c4
            ">python 3</span><span class="c6 ">and </span>
                <span class="c4 ">NumPy</span><span class="c6 ">&nbsp;on a </span><span class="c4 ">Windows</span><span class="c6 ">System.</span></p>
            <p class="c21 c13 "><span class="c6 ">&nbsp; I. &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span><span class="c43 c86 ">Python Installation</span></p>
            <p class="c21 c13 c48 "><span class="c6 ">1) &nbsp; &nbsp;First, download a 64 bit version of any 3.7.x from python.org. Clicking on the following link automatically downloads the required python</span></p>
            <p class="c21 c13 c139 "><span class="c0 ">https://www.python.org/ftp/python/3.7.2/python-3.7.2.exe</span></p>
            <p class="c21 c13 c48 "><span class="c6 ">2) &nbsp; &nbsp;Install the above file with &lt;</span><span class="c4 ">pip</span><span class="c6 ">&gt; support, and for that select custom installation and check all the boxes in the Optional Features window.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 456.50px;
            height: 288.97px; "><img alt=" " src="../images/image48.png " style="width: 456.50px; height: 288.97px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c21 c9 "><span class="c6 ">Note: don&#39;t forget to check (See next figure) &ldquo;</span><span class="c160 c58 c66 c55 c41 c165 ">Add Python to environment variables</span><span class="c6 ">&rdquo; for accessing python using the command prompt.</span></p>
            <p class="c21 c9 c7 "><span class="c80 c11 c58 "></span></p>
            <p class="c9 c94 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 482.50px;
            height: 294.60px; "><img alt=" " src="../images/image49.png " style="width: 482.50px; height: 294.60px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c94 c9 c7 "><span class="c80 c11 c58 "></span></p>
            <p class="c25 c13 "><span class="c58 c117 c55 c41 ">Python Modules Installations</span></p>
            <p class="c25 c13 c7 "><span class="c103 c42 c58 "></span></p>
            <p class="c25 c13 "><span class="c42 c58 ">After installing 3.X Python in Windows, you can simply open a command window (using Administrative privilege) and type:</span></p>
            <p class="c25 c13 c7 "><span class="c6 "></span></p>
            <p class="c25 c9 "><span class="c58 c41 c45 ">python -m pip install numpy</span></p>
            <p class="c25 c9 "><span class="c58 c41 c45 ">python -m pip install scipy</span></p>
            <p class="c25 c9 "><span class="c41 c45 c58 ">python -m pip install ase==3.17.0</span></p>
            <p class="c25 c9 "><span class="c58 c41 c45 ">python -m pip install matplotlib</span></p>
            <p class="c25 c13 c7 "><span class="c6 "></span></p>
            <p class="c25 c13 "><span class="c42 c58 ">To install </span><span class="c58 c55 c41 ">numpy, scipy, ase</span><span class="c42 c58 ">&nbsp;and </span><span class="c58 c55 c41 ">matplotlib </span><span class="c42 c58 ">libraries (note: it will install latest versions except for ASE).</span></p>
            <p class="c25 c13 c7 "><span class="c6 "></span></p>
            <p class="c25 c13 "><span class="c42 c58 ">(For a specific installation you can also do:&nbsp;&nbsp;</span></p>
            <p class="c13 c22 "><span class="c58 c66 c41 c45 ">python -m pip install nameOfPackage==x.y.z</span><span class="c42 c58 ">&nbsp;,</span></p>
            <p class="c25 c13 "><span class="c42 c58 ">&nbsp;where x,y,z are version numbers)</span></p>
            <p class="c94 c9 c7 "><span class="c80 c11 c58 "></span></p>
            <p class="c21 c13 "><span class="c6 ">&nbsp; II. &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span><span class="c43 c86 ">NumPy Installation</span></p>
            <p class="c25 c13 "><span class="c42 ">If the above command: </span><span class="c58 c41 c45 ">python -m pip install numpy </span><span class="c42 c58 ">fails, this method can be used to install the NumPy in Windows OS.</span></p>
            <p class="c21 c13 "><span class="c4 ">Disclaimer:</span><span class="c6 ">&nbsp;Since NumPy doesn&rsquo;t have an official build for windows, we&rsquo;ll be downloading an </span><span class="c4 ">unofficial version</span><span class="c6 ">&nbsp;from</span>
                <span class="c6 "><a class="c70 " href="https://www.google.com/url?q=https://www.lfd.uci.edu/&amp;sa=D&amp;source=editors&amp;ust=1612637412797000&amp;usg=AOvVaw1K0kkIpKZdcvjPte4JaQV- ">&nbsp;</a></span>
                <span class="c0 ">https://www.lfd.uci.edu/</span><span class="c6 ">&nbsp;which is maintained by Christoph Gohlke, Laboratory for Fluorescence Dynamics, University of California, Irvine.</span></p>
            <p class="c21 c13 "><span class="c6 ">1) &nbsp; &nbsp; Download the .whl file of NumPy using the link to your Desktop. &nbsp;Clicking on the following link automatically downloads the required python </span><span class="c0 ">https://download.lfd.uci.edu/pythonlibs/r5uhg2lo/numpy-1.16.1+mkl-cp37-cp37m-win_amd64.whl</span></p>
            <p class="c21 c13 "><span class="c6 ">If the file is not downloaded to the Desktop, Copy the &lt;.whl&gt; file to Desktop. </span></p>
            <p class="c21 c13 "><span class="c4 ">Note:</span><span class="c6 ">&nbsp;If the above link is not working, goto</span></p>
            <p class="c21 c13 "><span class="c0 ">https://www.lfd.uci.edu/~gohlke/pythonlibs/#numpy </span></p>
            <p class="c21 c13 "><span class="c6 ">and download the 1.16.1 version of </span><span class="c4 ">NumPy </span><span class="c6 ">(Not Numpy-quaternion).</span></p>
            <p class="c94 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 511.64px;
            height: 391.99px; "><img alt=" " src="../images/image40.png " style="width: 593.77px; height: 391.99px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c21 c13 "><span class="c6 ">2) Then run command prompt as administrator: Type &nbsp;&lt;cmd&gt; in start menu and right click on command prompt and click on &ldquo;</span><span class="c4
            ">Run as administrator</span><span class="c6 ">&rdquo; (See the next figure).</span></p>
            <p class="c21 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 487.67px;
            height: 268.00px; "><img alt=" " src="../images/image41.png " style="width: 512.01px; height: 396.06px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c21 c13 c7 "><span class="c80 c11 c58 "></span></p>
            <p class="c21 c13 c145 "><span class="c6 ">3) &nbsp; &nbsp; Navigate to your Desktop where the &lt;.whl&gt; file is saved. Type the following command in your command prompt and press Enter.</span></p>
            <p class="c21 c13 c78 "><span class="c46 c41 ">cd %systemdrive%\users\%username%\Desktop</span></p>
            <p class="c21 c13 c78 "><span class="c6 ">Or manually find your Desktop path and enter like:</span></p>
            <p class="c21 c13 c78 "><span class="c46 c41 ">cd C:\Users\staff\Desktop</span></p>
            <p class="c21 c13 c145 "><span class="c6 ">4) &nbsp; &nbsp; Install NumPy using the &lt;.whl&gt; file; Type this command in your command prompt and press Enter.</span></p>
            <p class="c21 c13 c78 "><span class="c46 c41 ">pip install &quot;numpy-1.16.0+mkl-cp37-cp37m-win_amd64.whl&quot;</span></p>
            <p class="c21 c13 c7 "><span class="c80 c11 c58 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <div id="appendix-c"></div>
            <p class="c65 c13 subtitle " id="h.z337ya "><span class="c6 ">Appendix-C</span></p>
            <p class="c2 c13 "><span class="c101 c36 c58 ">Windows Subsystem Linux in DJMol</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">For more advanced scripting/calculations one may want to use a real Linux operating system (instead of Cygwin emulators) in conjunction with DJMol. For this one can use WSL in the Windows 10 OS. It will give a real Linux OS and hence it guarantees a cent percentage of portability of C/C++ or Fortran codes into the Windows system.</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6
            ">Herein we give a short description of invoking of WSL in the DJMol platform (by assuming that you already have a WSL in your PC; otherwise it can be freely downloaded from Microsoft store, visit: &nbsp; </span><span class="c0 ">https://docs.microsoft.com/en-us/windows/wsl/install-win10</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c4 ">(a)&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Setting up the WSL</span></p>
            <p class="c2 c13 "><span class="c6 ">1. Update the Linux packages being used by WSL with the following commands in</span></p>
            <p class="c2 c13 "><span class="c6 ">your WSL Ubuntu command window:</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c27 c41 c45 ">sudo apt update</span></p>
            <p class="c2 c9 "><span class="c27 c41 c45 ">sudo apt upgrade</span></p>
            <p class="c2 c9 "><span class="c27 c41 c45 ">sudo apt install unzip</span></p>
            <p class="c2 c9 "><span class="c27 c41 c45 ">sudo apt install ssh</span></p>
            <p class="c2 c9 c7 "><span class="c27 c41 c45 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 527.33px;
            height: 113.33px; "><img alt=" " src="../images/image42.png " style="width: 527.33px; height: 293.34px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 "><span class="c6 ">2. Your Windows installation is probably already using port 22 for its SSH server, we need to change WSL&rsquo;s SSH server to listen to a different port.</span></p>
            <p class="c2 c13 c57 "><span class="c27 c41 c45 ">sudo nano /etc/ssh/sshd_config</span></p>
            <p class="c2 c13 c7 c57 "><span class="c27 c41 c45 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Change the lines:</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c3 "># What ports, IPs and protocols we listen for</span></p>
            <p class="c2 c13 c57 "><span class="c27 c41 c45 ">Port 22</span></p>
            <p class="c2 c13 "><span class="c6 ">to</span></p>
            <p class="c2 c13 c57 "><span class="c27 c41 c45 ">Port 2222</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 398.00px;
            height: 273.33px; "><img alt=" " src="../images/image43.png " style="width: 429.75px; height: 273.33px; margin-left: -10.40px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c20 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 414.00px;
            height: 293.33px; "><img alt=" " src="../images/image44.png " style="width: 457.85px; height: 293.33px; margin-left: -21.98px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 "><span class="c6 ">The WSL SSH server is initially set up to use key files for authentication. </span></p>
            <p class="c2 c13 "><span class="c6 ">To allow authentication with passwords, change:</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 "># Change to no to disable tunnelled clear text passwords</span></p>
            <p class="c2 c9 "><span class="c27 c41 c45 ">PasswordAuthentication no</span></p>
            <p class="c2 c13 "><span class="c27 c11 ">to</span></p>
            <p class="c2 c9 "><span class="c27 c41 c45 ">PasswordAuthentication yes</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Use </span><span class="c4 ">Ctrl-o</span><span class="c6 ">&nbsp;to save the file. Use </span><span class="c4 ">Ctrl-x</span><span class="c6 ">&nbsp;to exit the nano text editor.</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">3. Before using the WSL SSH server you must stop and restart the SSH server. You will perform this step every time that you use WSL as the remote host from NetBeans. I suggest placing the command in a script file that you can execute.</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c57 "><span class="c27 c41 c45 ">nano startssh.sh</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Insert the following text into the script file:</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c57 "><span class="c27 c41 c45 ">sudo service ssh --full-restart</span></p>
            <p class="c2 c13 c7 c57 "><span class="c27 c41 c45 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 482.67px;
            height: 186.67px; "><img alt=" " src="../images/image45.png " style="width: 482.67px; height: 186.67px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 "><span class="c6 ">Use</span><span class="c4 ">&nbsp;Ctrl-o</span><span class="c6 ">&nbsp;to save the file. Use </span><span class="c4 ">Ctrl-x</span><span class="c6 ">&nbsp;to exit the nano text editor.</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Back at the command line, use the following command to mark the script file as executable:</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c57 "><span class="c27 c41 c45 ">chmod a+x startssh.sh</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">To restart the SSH server, use the following command:</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c57 "><span class="c27 c41 c45 ">./startssh.sh</span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 418.67px;
            height: 239.33px; "><img alt=" " src="../images/image46.png " style="width: 418.67px; height: 239.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c4 ">(b)&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Setting up the Netbeans IDE</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">1. Open the Terminal and invoke the Terminal as:</span></p>
            <p class="c2 c13 c57 "><span class="c27 c41 c95 ">Window &rarr; IDE Tools &rarr; Terminal</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span class="c3 ">&nbsp;</span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad)
            translateZ(0px); width: 397.00px; height: 269.00px; "><img alt=" " src="../images/image36.png " style="width: 397.00px; height: 269.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad)
            translateZ(0px); " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">2. Create New Remote Terminal Tab.</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 435.00px;
            height: 228.00px; "><img alt=" " src="../images/image37.png " style="width: 435.00px; height: 228.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 "><span class="c6 ">3. Enter your credentials.</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c9 "><span class="c27 c41 c45 ">User : &lt;your-username&gt;</span></p>
            <p class="c2 c9 "><span class="c27 c41 c45 ">Host : localhost</span></p>
            <p class="c2 c9 "><span class="c27 c41 c45 ">SSH Port : 2222</span></p>
            <p class="c2 c9 c7 "><span class="c27 c41 c45 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 266.00px;
            height: 216.00px; "><img alt=" " src="../images/image38.png " style="width: 266.00px; height: 216.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">Click OK.</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">4. Choose Password.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 259.33px;
            height: 199.33px; "><img alt=" " src="../images/image39.png " style="width: 259.33px; height: 199.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c6 ">5. Enter your Linux Password.</span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 454.50px;
            height: 275.64px; "><img alt=" " src="../images/image71.png " style="width: 454.50px; height: 275.64px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 "><span class="c6 ">Click OK.</span></p>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 "><span class="c6 ">#Recommended: Check Remember Password box.</span></p>
            <p class="c2 c13 "><span class="c6 ">The process is finished; Now one can use WSL in DJMol.</span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c20 c13 image-container"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 483.00px;
            height: 253.00px; "><img alt=" " src="../images/image72.png " style="width: 483.00px; height: 253.00px; margin-left: -0.00px; margin-top: -0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);
            " title=" "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c3 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <div id="appendix-d"></div>
            <p class="c65 c13 subtitle " id="h.3j2qqm3 "><span class="c6 ">Appendix-D</span></p>
            <p class="c2 c13 "><span class="c101 c36 c58 ">Saved Data</span></p>
            <p class="c2 c13 c7 "><span class="c4 "></span></p>
            <p class="c2 c13 "><span class="c6 ">DJMol generate a number of text based </span><span class="c0 ">temporary </span><span class="c6 ">data, and these are nothing but a set of post-processing data from various visualizers or converters were temporarily stored in different directories and it can be used for a numerical comparison (for example, how MO levels of DFTB+ are different from that of Siesta for a given molecule). A list of such selected post processed files with its description is shown in the Table 1.</span></p>
            <a id="t.cc8210870adde45914c2a29744e7535351aab3c2 "></a>
            <a id="t.45 "></a>
            <table class="c44 ">
                <tbody>
                    <tr class="c15 ">
                        <td class="c126 " colspan="1 " rowspan="1 ">
                            <p class="c21 "><span class="c37 ">PESScan\2DPESscanmovie.xyz</span></p>
                            <p class="c21 "><span class="c37 ">PESScan\2DPESscan.dat</span></p>
                            <p class="c21 "><span class="c37 ">PESScan\View2DPES.dat</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\MODFTB.dat</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\MOSiesta.dat</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\ForceDFTB.dat</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\ForceSiesta.dat</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\SCFSiesta.txt</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\Uvdata_2.csv</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\Uvdata_1.csv</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\MDDFTBmovie.xyz</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\TotalMD.dat</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\KineticMD.dat</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\PotentialMD.dat</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\VelocityAC.out</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\DipoleAC.out</span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\StdOrientation.xyz</span></p>
                            <p class="c21 c7 "><span class="c37 "></span></p>
                            <p class="c21 "><span class="c37 ">SCRATCH\MullikenDFTB.dat</span></p>
                            <p class="c21 "><span class="c37 ">Input\dftb_in.hsd</span></p>
                            <p class="c21 "><span class="c37 ">Input\siestaTEMP.fdf</span></p>
                            <p class="c21 "><span class="c37 ">ModesBinary\modes_in.hsd</span></p>
                            <p class="c21 "><span class="c37 ">ModesBinary\waveplot_in.hsd</span></p>
                            <p class="c21 "><span class="c37 ">ModesBinary\modes.xyz &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p>
                            <p class="c21 "><span class="c37 ">ModesBinary\*.cube</span></p>
                            <p class="c21 "><span class="c37 ">SiestaApp\SiestaBinary.4.0CygWin64\*.cube</span></p>
                            <p class="c21 "><span class="c37 ">SiestaApp\SiestaBinary.4.0.CygWin64\SiestaVibModes.xyz</span></p>
                            <p class="c21 "><span class="c37 ">SiestaApp\SiestaBinary.4.0.CygWin64\BAND.bands</span></p>
                            <p class="c21 "><span class="c37 ">SiestaApp\SiestaBinary.4.0.CygWin64\wannier.bands</span></p>
                            <p class="c21 "><span class="c37 ">SiestaApp\SiestaBinary.4.0.CygWin64\wannier_*.xsf</span></p>
                            <p class="c21 "><span class="c37 ">OpenMD\.</span></p>
                            <p class="c21 "><span class="c37 ">Database\.</span></p>
                            <p class="c2 c13 c7 "><span class="c152 c42 c58 c113 "></span></p>
                        </td>
                        <td class="c150 " colspan="1 " rowspan="1 ">
                            <p class="c21 "><span class="c37 ">Geometries of molecules used for the PES</span></p>
                            <p class="c21 "><span class="c37 ">Energies and &nbsp;geometries of the PES</span></p>
                            <p class="c21 "><span class="c37 ">Coordinates (initial, final step size) and PES energy</span></p>
                            <p class="c21 "><span class="c37 ">MO energies in DFTB+</span></p>
                            <p class="c21 "><span class="c37 ">MO energies in Siesta</span></p>
                            <p class="c21 "><span class="c37 ">Cartesian Force components in DFTB+</span></p>
                            <p class="c21 "><span class="c37 ">Cartesian Force components in Siesta</span></p>
                            <p class="c21 "><span class="c37 ">SCF convergence in Siesta</span></p>
                            <p class="c21 "><span class="c37 ">Fitted UV-Vis data</span></p>
                            <p class="c21 "><span class="c37 ">UV-Vis oscillator strength for the spectrum</span></p>
                            <p class="c21 "><span class="c37 ">Trajectory of MD in XYZ animation format</span></p>
                            <p class="c21 "><span class="c37 ">Total energies of MD run</span></p>
                            <p class="c21 "><span class="c37 ">Kinetic energies of MD run</span></p>
                            <p class="c21 "><span class="c37 ">Potential energies of MD run</span></p>
                            <p class="c21 "><span class="c37 ">IR spectrum from MD using velocity autocorrelation</span></p>
                            <p class="c21 "><span class="c37 ">IR spectrum from MD using dipole moment autocorrelation</span></p>
                            <p class="c21 "><span class="c37 ">Standard orientation of the given molecule</span></p>
                            <p class="c21 c7 "><span class="c37 "></span></p>
                            <p class="c21 "><span class="c37 ">Mulliken charges from DFTB+</span></p>
                            <p class="c21 "><span class="c37 ">DFTB+ script writer out file</span></p>
                            <p class="c21 "><span class="c37 ">Siesta script writer out file</span></p>
                            <p class="c21 "><span class="c37 ">DFTB+ file used to construct vibrational data</span></p>
                            <p class="c21 "><span class="c37 ">DFTB+ file used to construct MO and its density data</span></p>
                            <p class="c21 "><span class="c37 ">DFTB vibrational modes and frequencies</span></p>
                            <p class="c21 "><span class="c37 ">DFTB Cube files for MOs and Densities</span></p>
                            <p class="c21 "><span class="c37 ">Siesta Cube files</span></p>
                            <p class="c21 "><span class="c37 ">Siesta vibrational modes and frequencies</span></p>
                            <p class="c21 "><span class="c37 ">Siesta (DFT) bands</span></p>
                            <p class="c21 "><span class="c37 ">Siesta Wannier bands</span></p>
                            <p class="c129 "><span class="c37 ">Siesta Wannier orbitals</span></p>
                            <p class="c129 "><span class="c37 ">All OpenMD in and out files</span></p>
                            <p class="c18 c13 "><span class="c37 ">All downloaded Structural files from online repositories</span></p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p class="c2 c13 c7 "><span class="c6 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c21 c13 c7 "><span class="c3 "></span></p>
            <p class="c21 c13 "><span class="c3 ">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></p>
            <p class="c21 c13 "><span class="c3 ">&nbsp; &nbsp;</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <div id="appendix-e"></div>
            <p class="c65 c13 subtitle "><span class="c6 ">Appendix-E</span></p>
            <p class="c2 c13 "><span class="c101 c36 c58 ">Demonstration Videos</span></p>
            <p class="c2 c13 "><span class="c63 c42 c58 ">Please visit: </span><span class="c42 c58 c66 c142 ">https://www.youtube.com/channel/UCNczegqwli6gnuo6eqNJSPg</span><span class="c63 c42 c58 ">&nbsp;for Demo videos. The list is (as of October 2020):</span></p>
            <p class="c33 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Windows OpenMP settings | Executing Stand-alone DFTB+</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DJMol Demo on Point Group Symmetry Detection</span></p>
            <p class="c18 c13 "><span class="c61 c42 c74 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Running a DFTB Calculation from DJMol</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Phonon Density of States (DOS) of Aluminium Bulk: DJMol + ASE Python Scripting</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;PES demo in DJMol (Linux)</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Demo on Molecular Vibration of Water Molecule</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Demo on Partial Charges And File Conversions</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Generating/Displaying Molecular Orbitals in DJMol</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DFTB+ Input script Writer in DJMol</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DJmol/Netbeans : How to install ASE, Matplotlib with PIP</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adding Python Plugin in Netbeans or in DJMol</span></p>
            <p class="c18 c13 "><span class="c61 c42 c74 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Siesta&#39;s Charge density and/or electronic wave functions: 2D Contours/Surfaces</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Siesta&#39;s Charge density and/or electronic wave functions: 3D cube files</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Remote Submission using SSH tool in DJMol (64v)</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Band diagram and DOS (density of state) plot of Aluminum FCC</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Molecular Dynamics Analysis</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;UV-visible spectrum in DJMol</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c152 c86 c109 c107 c160 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Z-Matrix Structure Editor of DJMol</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Wannier AddOn in DJMol (with Siesta)</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DJMol Win64 Installation from its ZIP distribution File</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DJMol Version Control with Github</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Terminal use in DJMol</span></p>
            <p class="c18 c13 "><span class="c6 ">&nbsp;</span><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Building of DJMol with Netbeans 8.2 (Win64OS)</span></p>
            <p class="c18 c13 "><span class="c152 c42 c160 c109 ">&nbsp;</span><span class="c152 c160 c86 c109 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Radial Distribution Function from ASAP MD Calculation in the DJMol System</span></p>
            <p class="c18 c13 "><span class="c61 c74 c86 c107 ">&#9654;</span><span class="c6 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Burning of iso-octane fuel at 2500 K (molecular dynamics with DFTB+).</span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <div id="appendix-f"></div>
            <p class="c65 c13 subtitle "><span class="c6 ">Appendix-F</span></p>
            <p class="c18 c13 "><span class="c101 c36 c58 ">For Compiling/Installation of DJMol and Add-Ons</span></p>
            <p class="c2 c13 "><span class="c4 ">General Information</span></p>
            <p class="c13 c18 "><span>A familiarity with programming using Netbeans IDE is assumed. All of the DJMol Program was compiled by using </span><span class="c66 ">Netbeans IDE, v 8.2 (64bit)</span><span>&nbsp;in &nbsp;Windows 64 OS. This program and all other mandate packages for compiling the software are available at free of cost and to download these packages please see: </span>
                <span class="c66 ">http://www.djmol.info/download.html</span>
            </p>
            <p class="c18 c13 c7 "><span class="c3 "></span></p>
            <p class="c18 c13 "><span class="c3 ">The needed Packages are:</span></p>
            <p class="c18 c13 c7 "><span class="c3 "></span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">JAVA 1.8</span><span class="c5 ">&nbsp;(jdk-8u201-windows-x64.exe)</span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">Netbeans 8.2</span><span class="c5 ">&nbsp;(netbeans-8.2-javaee-windows.exe)</span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">Python plug-in</span><span class="c5 ">&nbsp;(2017-08-29-nbpython-nbms.zip)</span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">Python 3.7.2</span><span class="c5 ">&nbsp;(python-3.7.2-amd64.exe)</span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">Pip script</span><span class="c5 ">&nbsp;(get-pip.py)</span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">Numpy</span><span class="c5 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(numpy-1.16.0+mkl-cp37-cp37m-win_amd64.whl)</span></p>
            <p class="c18 c13 c7 "><span class="c5 "></span></p>
            <p class="c18 c13 "><span class="c3 ">We also used (64 bit) the following Python packages with version numbers:</span></p>
            <p class="c18 c13 c7 "><span class="c3 "></span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">SciPy</span><span class="c5 ">&nbsp;(scipy 1.2.1) &nbsp;</span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">ASE</span><span class="c5 ">&nbsp;(ASE 3.17.0)</span></p>
            <p class="c18 c13 "><span class="c34 ">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c41 c45 ">Matplotlib</span><span class="c5 ">&nbsp;(Matplotlib 3.0.3)</span></p>
            <p class="c18 c13 "><span class="c3 ">And these programs can be downloaded by using pip (use pip3) using a command window.</span></p>
            <p class="c2 c13 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c27 c77 c41 ">[A] Compiling Main Program, &lt;DJMolplatform&gt;</span></p>
            <p class="c2 c9 "><span>[1] Open the Project Directory, &lt;DJmol Platform v2.1&gt; and </span><span class="c41 ">Compile</span><span>&nbsp;and </span><span class="c41 ">Build</span><span class="c3 ">&nbsp;DJMol program. See the YouTube channel for its demonstration.</span></p>
            <p class="c2 c9 "><span>[2] Then make a Zip Distribution (by Selecting project &lt;DJmol Platform v2.1&gt; followed by selecting &nbsp;&lt;</span><span class="c41 ">Package as</span><span>&gt;, &lt;</span><span class="c41
            ">Zip Distribution</span><span class="c3 ">&gt;). A new &lt;dist&gt; folder will be created and it holds the Zip file.</span></p>
            <p class="c2 c9 c7 "><span class="c3 "></span></p>
            <p class="c2 c13 "><span class="c27 c77 c41 ">[B] Installing Program, &lt;DJMolplatform&gt;</span></p>
            <p class="c2 c9 "><span>[3] Go to &lt;</span><span class="c41 ">dist</span><span>&gt; folder and unzip the file and Copy </span><span class="c41 c168 ">all</span><span>&nbsp;the files/folders from the &lt;</span><span class="c41 ">Auxiliary</span><span>&gt; folder into &lt;</span>
                <span class="c41 ">dist\djmolplatform1</span><span class="c3 ">&gt;. This is shown in the demonstration.</span></p>
            <p class="c2 c9 "><span>[4] Go to &lt;</span><span class="c41 ">dist\djmolplatform1\etc</span><span>&gt; and replace the following line in the &lt;</span><span class="c41 ">djmolplatform1.CONF</span><span class="c3 ">&gt; file:</span></p>
            <p class="c2 c13 "><span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c34 c113 ">default_options=&quot;--branding djmolplatform1 -J-Xms24m -J-Xmx64m&quot;</span><span class="c3 ">&nbsp; &nbsp; to</span></p>
            <p class="c2 c13 "><span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span><span class="c26 ">default_options=&quot;--branding djmolplatform1 -J-Xms240m -J-Xmx640m&quot;</span></p>
            <p class="c2 c9 "><span>[5] To adjust the </span><span class="c41 ">Resolution</span><span>&nbsp;of the program &nbsp;Right click on &nbsp;&lt;</span><span class="c41
            ">djmolplatform164.exe</span><span>&gt; then move to &nbsp; &lt;</span><span class="c41 ">Compatibility</span><span>&gt; tab and click on &lt;</span>
                <span class="c41 ">Change high DPI settings</span><span>&gt; and Select the option &quot;</span><span class="c41 ">Override high DPI scaling behavior</span><span class="c3 ">&quot;.</span></p>
            <p class="c2 c13 c57 "><span>[6] Follow the instruction in the </span><span class="c41 ">PathVariablesSetting.pdf</span><span class="c3 ">&nbsp;file.</span></p>
            <p class="c2 c13 "><span>DJMol program can be now simply executed by double clicking &lt;</span><span class="c41 ">djmolplatform164.exe</span><span>&gt; which is located in &lt;</span><span class="c41 ">bin</span><span class="c3 ">&gt; folder. &nbsp;</span></p>
            <p class="c2 c13 "><span>Note that for the First Time of the execution, you may want to select, &nbsp;&lt;</span><span class="c41 ">Disable Modules and Continue</span><span>&gt; option. To fix this, you can try to clean your user-directory as it is mentioned in </span>
                <span class="c168 c170 "><a class="c70 " href="https://www.google.com/url?q=http://wiki.netbeans.org/FaqWhatIsUserdir&amp;sa=D&amp;source=editors&amp;ust=1612637412828000&amp;usg=AOvVaw3f6zpw0eAYiy0UDyg1SqdX ">http://wiki.netbeans.org/FaqWhatIsUserdir</a></span>
            </p>
            <p class="c2 c13 "><span class="c3 ">And once the application starts do the following (not mandate but it is useful):</span></p>
            <p class="c13 c23 "><span class="c27 c41 c45 ">Close &lt;Start Page&gt;</span></p>
            <p class="c23 c13 "><span class="c27 c41 c45 ">Close &lt;Services&gt;</span></p>
            <p class="c23 c13 "><span class="c27 c41 c45 ">Add &lt;Windows -&gt; Favorites&gt;</span></p>
            <p class="c23 c13 "><span class="c27 c41 c45 ">Add &lt;Windows -&gt; Output&gt;</span></p>
            <p class="c2 c13 "><span class="c27 c77 c41 ">[B] Compiling Add-On Programs</span></p>
            <p class="c2 c13 c57 "><span class="c3 ">See &lt;AddOns_compilation.txt&gt; in Add-On directory.</span></p>
            <p class="c2 c13 c7 c57 "><span class="c3 "></span></p>
            <p class="c2 c13 c7 c57 "><span class="c3 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <div id="appendix-g"></div>
            <p class="c65 c13 subtitle "><span class="c6 ">Appendix-G</span></p>
            <p class="c18 c13 "><span class="c101 c36 c58 ">Forthcoming Features &nbsp; &nbsp; </span></p>
            <p class="c18 c13 c7 "><span class="c36 c58 c101 "></span></p>
            <p class="c18 c13 "><span class="c3 ">In this version DFTB+ 17.1 version is used and it will be replaced by DFTB+ 20.0 in the next version of the DJMol. Apart from this, other major components that would be a part of the program are:</span></p>
            <p class="c18 c13 c7 "><span class="c3 "></span></p>
            <ul class="c112 lst-kix_list_9-0 start ">
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">Huckel (for hydrocarbons or Fullerenes) and Extended Huckel theory (EHT) with analytical overlap integrals. Molecular electronics module using EHT will also be included</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">IR and Raman Numerical intensities for DFTB+ module</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">Vibrational mode designations for DFTB+ modes</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">Projection of Translational and Rotational modes from the Siesta/DFTB+ Hessian</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">More DJMol Exclusive or GPL Python Scripts/Utilities</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">KisTHELP module to compute reaction rates</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">PASTA (developed by IKST, Bangalore) module to locate transition state etc.</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">NWChem add-on for quantum chemistry (unlike GAMESS it is GPL and support Gaussian as well as planewave basis set)</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">Java based CDK (Chemistry Development Kit) for cheminformatics</span></li>
                <li class="c18 c9 c59 li-bullet-0 "><span class="c6 ">JShell support</span></li>
            </ul>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <p class="c18 c13 c7 "><span class="c6 "></span></p>
            <div>
                <p class="c13 c7 c67 "><span class="c3 "></span></p>
                <p class="c33 c13 c7 "><span class="c3 "></span></p>
            </div>
        </div>
    </div>
    <button id="myBtn"><a href="#top" style="color: white">Top</a></button>
</body>

</html>